// @mirohq/miro-api/dist/model/appCardDataPlatformTags.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppCardDataPlatformTags = void 0;
/**
 * @internal
 * Contains app card item data, such as the title, description, or fields.
 */
class AppCardDataPlatformTags {
    /** @ignore */
    static getAttributeTypeMap() {
        return AppCardDataPlatformTags.attributeTypeMap;
    }
}
exports.AppCardDataPlatformTags = AppCardDataPlatformTags;
/** @ignore */
AppCardDataPlatformTags.discriminator = undefined;
/** @ignore */
AppCardDataPlatformTags.attributeTypeMap = [
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'fields',
        baseName: 'fields',
        type: 'Array<CustomField>',
    },
    {
        name: 'owned',
        baseName: 'owned',
        type: 'boolean',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'AppCardDataPlatformTags.StatusEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
];
(function (AppCardDataPlatformTags) {
    AppCardDataPlatformTags.StatusEnum = {
        Disconnected: 'disconnected',
        Connected: 'connected',
        Disabled: 'disabled',
    };
})(AppCardDataPlatformTags = exports.AppCardDataPlatformTags || (exports.AppCardDataPlatformTags = {}));
