// @mirohq/miro-api/dist/model/appCardStylePlatformBulkCreateOperationExperimentalRelease.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppCardStylePlatformBulkCreateOperationExperimentalRelease = void 0;
/**
 * @internal
 * Contains information about the style of an app card item, such as the fill color.
 */
class AppCardStylePlatformBulkCreateOperationExperimentalRelease {
    /** @ignore */
    static getAttributeTypeMap() {
        return AppCardStylePlatformBulkCreateOperationExperimentalRelease.attributeTypeMap;
    }
}
exports.AppCardStylePlatformBulkCreateOperationExperimentalRelease = AppCardStylePlatformBulkCreateOperationExperimentalRelease;
/** @ignore */
AppCardStylePlatformBulkCreateOperationExperimentalRelease.discriminator = undefined;
/** @ignore */
AppCardStylePlatformBulkCreateOperationExperimentalRelease.attributeTypeMap = [
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'string',
    },
];
