// @mirohq/miro-api/dist/model/appCardDataChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppCardDataChanges = void 0;
/**
 * @internal
 * Contains app card item data, such as the title, description, or fields.
 */
class AppCardDataChanges {
    constructor() {
        /**
         * Status indicating whether an app card is connected and in sync with the source. When the source for the app card is deleted, the status returns `disabled`.
         */
        this['status'] = AppCardDataChanges.StatusEnum.Disconnected;
        /**
         * A short text header to identify the app card.
         */
        this['title'] = 'sample app card item';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return AppCardDataChanges.attributeTypeMap;
    }
}
exports.AppCardDataChanges = AppCardDataChanges;
/** @ignore */
AppCardDataChanges.discriminator = undefined;
/** @ignore */
AppCardDataChanges.attributeTypeMap = [
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'fields',
        baseName: 'fields',
        type: 'Array<CustomField>',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'AppCardDataChanges.StatusEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
];
(function (AppCardDataChanges) {
    AppCardDataChanges.StatusEnum = {
        Disconnected: 'disconnected',
        Connected: 'connected',
        Disabled: 'disabled',
    };
})(AppCardDataChanges = exports.AppCardDataChanges || (exports.AppCardDataChanges = {}));
