// @mirohq/miro-api/dist/model/appCardCreateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppCardCreateRequest = void 0;
class AppCardCreateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return AppCardCreateRequest.attributeTypeMap;
    }
}
exports.AppCardCreateRequest = AppCardCreateRequest;
/** @ignore */
AppCardCreateRequest.discriminator = undefined;
/** @ignore */
AppCardCreateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'AppCardDataChanges',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'AppCardStyle',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'Geometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
