// @mirohq/miro-api/dist/model/appCardDataResponse.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppCardDataResponse = void 0;
/**
 * @internal
 * Contains app card item data, such as the title, description, or fields.
 */
class AppCardDataResponse {
    /** @ignore */
    static getAttributeTypeMap() {
        return AppCardDataResponse.attributeTypeMap;
    }
}
exports.AppCardDataResponse = AppCardDataResponse;
/** @ignore */
AppCardDataResponse.discriminator = undefined;
/** @ignore */
AppCardDataResponse.attributeTypeMap = [
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'fields',
        baseName: 'fields',
        type: 'Array<CustomField>',
    },
    {
        name: 'owned',
        baseName: 'owned',
        type: 'boolean',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'AppCardDataResponse.StatusEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
];
(function (AppCardDataResponse) {
    AppCardDataResponse.StatusEnum = {
        Disconnected: 'disconnected',
        Connected: 'connected',
        Disabled: 'disabled',
    };
})(AppCardDataResponse = exports.AppCardDataResponse || (exports.AppCardDataResponse = {}));
