// @mirohq/miro-api/dist/model/basicErrorOrganizationsEnterprisePlan.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasicErrorOrganizationsEnterprisePlan = void 0;
class BasicErrorOrganizationsEnterprisePlan {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'error';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return BasicErrorOrganizationsEnterprisePlan.attributeTypeMap;
    }
}
exports.BasicErrorOrganizationsEnterprisePlan = BasicErrorOrganizationsEnterprisePlan;
/** @ignore */
BasicErrorOrganizationsEnterprisePlan.discriminator = undefined;
/** @ignore */
BasicErrorOrganizationsEnterprisePlan.attributeTypeMap = [
    {
        name: 'status',
        baseName: 'status',
        type: 'number',
    },
    {
        name: 'code',
        baseName: 'code',
        type: 'string',
    },
    {
        name: 'message',
        baseName: 'message',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
