// @mirohq/miro-api/dist/model/basicError.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasicError = void 0;
class BasicError {
    constructor() {
        this['type'] = 'error';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return BasicError.attributeTypeMap;
    }
}
exports.BasicError = BasicError;
/** @ignore */
BasicError.discriminator = undefined;
/** @ignore */
BasicError.attributeTypeMap = [
    {
        name: 'status',
        baseName: 'status',
        type: 'number',
    },
    {
        name: 'code',
        baseName: 'code',
        type: 'string',
    },
    {
        name: 'message',
        baseName: 'message',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
