// @mirohq/miro-api/dist/model/boardDataClassificationLabel.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoardDataClassificationLabel = void 0;
class BoardDataClassificationLabel {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'board-data-classification-label';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return BoardDataClassificationLabel.attributeTypeMap;
    }
}
exports.BoardDataClassificationLabel = BoardDataClassificationLabel;
/** @ignore */
BoardDataClassificationLabel.discriminator = undefined;
/** @ignore */
BoardDataClassificationLabel.attributeTypeMap = [
    {
        name: 'color',
        baseName: 'color',
        type: 'string',
    },
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'name',
        baseName: 'name',
        type: 'string',
    },
    {
        name: 'sharingRecommendation',
        baseName: 'sharingRecommendation',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
