// @mirohq/miro-api/dist/model/boardMember.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoardMember = void 0;
/**
 * @internal
 * Contains the current user\'s board membership details. The current user could be different from the board owner.
 */
class BoardMember {
    /** @ignore */
    static getAttributeTypeMap() {
        return BoardMember.attributeTypeMap;
    }
}
exports.BoardMember = BoardMember;
/** @ignore */
BoardMember.discriminator = undefined;
/** @ignore */
BoardMember.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'name',
        baseName: 'name',
        type: 'string',
    },
    {
        name: 'role',
        baseName: 'role',
        type: 'BoardMember.RoleEnum',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
(function (BoardMember) {
    BoardMember.RoleEnum = {
        Viewer: 'viewer',
        Commenter: 'commenter',
        Editor: 'editor',
        Coowner: 'coowner',
        Owner: 'owner',
    };
})(BoardMember = exports.BoardMember || (exports.BoardMember = {}));
