// @mirohq/miro-api/dist/model/boardSubscriptionData.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoardSubscriptionData = void 0;
/**
 * @internal
 * Contains information about a webhook subscription, such as the board ID associated with the webhook subscription, the date and time when the webhook subscription was last updated, and the type of board item that the subscription is associated with.
 */
class BoardSubscriptionData {
    /** @ignore */
    static getAttributeTypeMap() {
        return BoardSubscriptionData.attributeTypeMap;
    }
}
exports.BoardSubscriptionData = BoardSubscriptionData;
/** @ignore */
BoardSubscriptionData.discriminator = undefined;
/** @ignore */
BoardSubscriptionData.attributeTypeMap = [
    {
        name: 'boardId',
        baseName: 'boardId',
        type: 'string',
    },
];
