// @mirohq/miro-api/dist/model/boardPermissionsPolicy.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoardPermissionsPolicy = void 0;
/**
 * @internal
 * Defines the permissions policies for the board.
 */
class BoardPermissionsPolicy {
    constructor() {
        /**
         * Defines who can start or stop timer, voting, video chat, screen sharing, attention management. Others will only be able to join. To change the value for the `collaborationToolsStartAccess` parameter, contact Miro Customer Support.
         */
        this['collaborationToolsStartAccess'] = BoardPermissionsPolicy.CollaborationToolsStartAccessEnum.AllEditors;
        /**
         * Defines who can copy the board, copy objects, download images, and save the board as a template or PDF.
         */
        this['copyAccess'] = BoardPermissionsPolicy.CopyAccessEnum.Anyone;
        /**
         * Defines who can change access and invite users to this board. To change the value for the `sharingAccess` parameter, contact Miro Customer Support.
         */
        this['sharingAccess'] = BoardPermissionsPolicy.SharingAccessEnum.TeamMembersWithEditingRights;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return BoardPermissionsPolicy.attributeTypeMap;
    }
}
exports.BoardPermissionsPolicy = BoardPermissionsPolicy;
/** @ignore */
BoardPermissionsPolicy.discriminator = undefined;
/** @ignore */
BoardPermissionsPolicy.attributeTypeMap = [
    {
        name: 'collaborationToolsStartAccess',
        baseName: 'collaborationToolsStartAccess',
        type: 'BoardPermissionsPolicy.CollaborationToolsStartAccessEnum',
    },
    {
        name: 'copyAccess',
        baseName: 'copyAccess',
        type: 'BoardPermissionsPolicy.CopyAccessEnum',
    },
    {
        name: 'sharingAccess',
        baseName: 'sharingAccess',
        type: 'BoardPermissionsPolicy.SharingAccessEnum',
    },
];
(function (BoardPermissionsPolicy) {
    BoardPermissionsPolicy.CollaborationToolsStartAccessEnum = {
        AllEditors: 'all_editors',
        BoardOwnersAndCoowners: 'board_owners_and_coowners',
    };
    BoardPermissionsPolicy.CopyAccessEnum = {
        Anyone: 'anyone',
        TeamMembers: 'team_members',
        TeamEditors: 'team_editors',
        BoardOwner: 'board_owner',
    };
    BoardPermissionsPolicy.SharingAccessEnum = {
        TeamMembersWithEditingRights: 'team_members_with_editing_rights',
        OwnerAndCoowners: 'owner_and_coowners',
    };
})(BoardPermissionsPolicy = exports.BoardPermissionsPolicy || (exports.BoardPermissionsPolicy = {}));
