// @mirohq/miro-api/dist/model/boardSharingPolicy.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoardSharingPolicy = void 0;
/**
 * @internal
 * Defines the public-level, organization-level, and team-level access for the board. The access level that a user gets depends on the highest level of access that results from considering the public-level, team-level, organization-level, and direct sharing access.
 */
class BoardSharingPolicy {
    constructor() {
        /**
         * Defines the user role when inviting a user via the invite to team and board link. For Enterprise users, the `inviteToAccountAndBoardLinkAccess` parameter is always set to `no_access`.
         */
        this['inviteToAccountAndBoardLinkAccess'] = BoardSharingPolicy.InviteToAccountAndBoardLinkAccessEnum.NoAccess;
        /**
         * Defines the organization-level access to the board. If the board is created for a team that does not belong to an organization, the `organizationAccess` parameter is always set to the default value.
         */
        this['organizationAccess'] = BoardSharingPolicy.OrganizationAccessEnum.Private;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return BoardSharingPolicy.attributeTypeMap;
    }
}
exports.BoardSharingPolicy = BoardSharingPolicy;
/** @ignore */
BoardSharingPolicy.discriminator = undefined;
/** @ignore */
BoardSharingPolicy.attributeTypeMap = [
    {
        name: 'access',
        baseName: 'access',
        type: 'BoardSharingPolicy.AccessEnum',
    },
    {
        name: 'inviteToAccountAndBoardLinkAccess',
        baseName: 'inviteToAccountAndBoardLinkAccess',
        type: 'BoardSharingPolicy.InviteToAccountAndBoardLinkAccessEnum',
    },
    {
        name: 'organizationAccess',
        baseName: 'organizationAccess',
        type: 'BoardSharingPolicy.OrganizationAccessEnum',
    },
    {
        name: 'teamAccess',
        baseName: 'teamAccess',
        type: 'BoardSharingPolicy.TeamAccessEnum',
    },
];
(function (BoardSharingPolicy) {
    BoardSharingPolicy.AccessEnum = {
        Private: 'private',
        View: 'view',
        Edit: 'edit',
        Comment: 'comment',
    };
    BoardSharingPolicy.InviteToAccountAndBoardLinkAccessEnum = {
        Viewer: 'viewer',
        Commenter: 'commenter',
        Editor: 'editor',
        Coowner: 'coowner',
        Owner: 'owner',
        Guest: 'guest',
        NoAccess: 'no_access',
    };
    BoardSharingPolicy.OrganizationAccessEnum = {
        Private: 'private',
        View: 'view',
        Comment: 'comment',
        Edit: 'edit',
    };
    BoardSharingPolicy.TeamAccessEnum = {
        Private: 'private',
        View: 'view',
        Comment: 'comment',
        Edit: 'edit',
    };
})(BoardSharingPolicy = exports.BoardSharingPolicy || (exports.BoardSharingPolicy = {}));
