// @mirohq/miro-api/dist/model/boardMembersInvite.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoardMembersInvite = void 0;
class BoardMembersInvite {
    constructor() {
        /**
         * Role of the board member.
         */
        this['role'] = BoardMembersInvite.RoleEnum.Commenter;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return BoardMembersInvite.attributeTypeMap;
    }
}
exports.BoardMembersInvite = BoardMembersInvite;
/** @ignore */
BoardMembersInvite.discriminator = undefined;
/** @ignore */
BoardMembersInvite.attributeTypeMap = [
    {
        name: 'emails',
        baseName: 'emails',
        type: 'Array<string>',
    },
    {
        name: 'role',
        baseName: 'role',
        type: 'BoardMembersInvite.RoleEnum',
    },
    {
        name: 'message',
        baseName: 'message',
        type: 'string',
    },
];
(function (BoardMembersInvite) {
    BoardMembersInvite.RoleEnum = {
        Viewer: 'viewer',
        Commenter: 'commenter',
        Editor: 'editor',
        Coowner: 'coowner',
        Owner: 'owner',
    };
})(BoardMembersInvite = exports.BoardMembersInvite || (exports.BoardMembersInvite = {}));
