// @mirohq/miro-api/dist/model/board.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Board = void 0;
/**
 * @internal
 * Contains the result data.
 */
class Board {
    /** @ignore */
    static getAttributeTypeMap() {
        return Board.attributeTypeMap;
    }
}
exports.Board = Board;
/** @ignore */
Board.discriminator = undefined;
/** @ignore */
Board.attributeTypeMap = [
    {
        name: 'createdAt',
        baseName: 'createdAt',
        type: 'Date',
    },
    {
        name: 'createdBy',
        baseName: 'createdBy',
        type: 'UserInfoShort',
    },
    {
        name: 'currentUserMembership',
        baseName: 'currentUserMembership',
        type: 'BoardMember',
    },
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'modifiedAt',
        baseName: 'modifiedAt',
        type: 'Date',
    },
    {
        name: 'modifiedBy',
        baseName: 'modifiedBy',
        type: 'UserInfoShort',
    },
    {
        name: 'name',
        baseName: 'name',
        type: 'string',
    },
    {
        name: 'owner',
        baseName: 'owner',
        type: 'UserInfoShort',
    },
    {
        name: 'picture',
        baseName: 'picture',
        type: 'Picture',
    },
    {
        name: 'policy',
        baseName: 'policy',
        type: 'BoardPolicy',
    },
    {
        name: 'team',
        baseName: 'team',
        type: 'Team',
    },
    {
        name: 'project',
        baseName: 'project',
        type: 'Project',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
    {
        name: 'viewLink',
        baseName: 'viewLink',
        type: 'string',
    },
];
