// @mirohq/miro-api/dist/model/boardMemberChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoardMemberChanges = void 0;
class BoardMemberChanges {
    constructor() {
        /**
         * Role of the board member.
         */
        this['role'] = BoardMemberChanges.RoleEnum.Commenter;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return BoardMemberChanges.attributeTypeMap;
    }
}
exports.BoardMemberChanges = BoardMemberChanges;
/** @ignore */
BoardMemberChanges.discriminator = undefined;
/** @ignore */
BoardMemberChanges.attributeTypeMap = [
    {
        name: 'role',
        baseName: 'role',
        type: 'BoardMemberChanges.RoleEnum',
    },
];
(function (BoardMemberChanges) {
    BoardMemberChanges.RoleEnum = {
        Viewer: 'viewer',
        Commenter: 'commenter',
        Editor: 'editor',
        Coowner: 'coowner',
        Owner: 'owner',
    };
})(BoardMemberChanges = exports.BoardMemberChanges || (exports.BoardMemberChanges = {}));
