// @mirohq/miro-api/dist/model/boardExportTaskResult.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoardExportTaskResult = void 0;
/**
 * @internal
 * Board export task results.
 */
class BoardExportTaskResult {
    /** @ignore */
    static getAttributeTypeMap() {
        return BoardExportTaskResult.attributeTypeMap;
    }
}
exports.BoardExportTaskResult = BoardExportTaskResult;
/** @ignore */
BoardExportTaskResult.discriminator = undefined;
/** @ignore */
BoardExportTaskResult.attributeTypeMap = [
    {
        name: 'boardId',
        baseName: 'boardId',
        type: 'string',
    },
    {
        name: 'errorMessage',
        baseName: 'errorMessage',
        type: 'string',
    },
    {
        name: 'exportLink',
        baseName: 'exportLink',
        type: 'string',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'string',
    },
];
