// @mirohq/miro-api/dist/model/boardMemberWithLinks.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoardMemberWithLinks = void 0;
class BoardMemberWithLinks {
    /** @ignore */
    static getAttributeTypeMap() {
        return BoardMemberWithLinks.attributeTypeMap;
    }
}
exports.BoardMemberWithLinks = BoardMemberWithLinks;
/** @ignore */
BoardMemberWithLinks.discriminator = undefined;
/** @ignore */
BoardMemberWithLinks.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'name',
        baseName: 'name',
        type: 'string',
    },
    {
        name: 'role',
        baseName: 'role',
        type: 'BoardMemberWithLinks.RoleEnum',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'SelfLink',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
(function (BoardMemberWithLinks) {
    BoardMemberWithLinks.RoleEnum = {
        Viewer: 'viewer',
        Commenter: 'commenter',
        Editor: 'editor',
        Coowner: 'coowner',
        Owner: 'owner',
    };
})(BoardMemberWithLinks = exports.BoardMemberWithLinks || (exports.BoardMemberWithLinks = {}));
