// @mirohq/miro-api/dist/model/boardPolicyChange.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoardPolicyChange = void 0;
/**
 * @internal
 * Defines the permissions policies and sharing policies for the board.
 */
class BoardPolicyChange {
    /** @ignore */
    static getAttributeTypeMap() {
        return BoardPolicyChange.attributeTypeMap;
    }
}
exports.BoardPolicyChange = BoardPolicyChange;
/** @ignore */
BoardPolicyChange.discriminator = undefined;
/** @ignore */
BoardPolicyChange.attributeTypeMap = [
    {
        name: 'permissionsPolicy',
        baseName: 'permissionsPolicy',
        type: 'BoardPermissionsPolicy',
    },
    {
        name: 'sharingPolicy',
        baseName: 'sharingPolicy',
        type: 'BoardSharingPolicyChange',
    },
];
