// @mirohq/miro-api/dist/model/bulkOperationError.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BulkOperationError = void 0;
/**
 * @internal
 * Error information with details about operation failure
 */
class BulkOperationError {
    /** @ignore */
    static getAttributeTypeMap() {
        return BulkOperationError.attributeTypeMap;
    }
}
exports.BulkOperationError = BulkOperationError;
/** @ignore */
BulkOperationError.discriminator = undefined;
/** @ignore */
BulkOperationError.attributeTypeMap = [
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
    {
        name: 'code',
        baseName: 'code',
        type: 'string',
    },
    {
        name: 'message',
        baseName: 'message',
        type: 'string',
    },
    {
        name: 'context',
        baseName: 'context',
        type: 'BulkOperationErrorContext',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'number',
    },
];
