// @mirohq/miro-api/dist/model/caption.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Caption = void 0;
/**
 * @internal
 * Contains the connector\'s caption data, such as content and its position.
 */
class Caption {
    /** @ignore */
    static getAttributeTypeMap() {
        return Caption.attributeTypeMap;
    }
}
exports.Caption = Caption;
/** @ignore */
Caption.discriminator = undefined;
/** @ignore */
Caption.attributeTypeMap = [
    {
        name: 'content',
        baseName: 'content',
        type: 'string',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'string',
    },
    {
        name: 'textAlignVertical',
        baseName: 'textAlignVertical',
        type: 'Caption.TextAlignVerticalEnum',
    },
];
(function (Caption) {
    Caption.TextAlignVerticalEnum = {
        Top: 'top',
        Middle: 'middle',
        Bottom: 'bottom',
    };
})(Caption = exports.Caption || (exports.Caption = {}));
