// @mirohq/miro-api/dist/model/cardStylePlatformBulkCreateOperationExperimentalRelease.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardStylePlatformBulkCreateOperationExperimentalRelease = void 0;
/**
 * @internal
 * Contains information about the style of a card item, such as the card theme.
 */
class CardStylePlatformBulkCreateOperationExperimentalRelease {
    /** @ignore */
    static getAttributeTypeMap() {
        return CardStylePlatformBulkCreateOperationExperimentalRelease.attributeTypeMap;
    }
}
exports.CardStylePlatformBulkCreateOperationExperimentalRelease = CardStylePlatformBulkCreateOperationExperimentalRelease;
/** @ignore */
CardStylePlatformBulkCreateOperationExperimentalRelease.discriminator = undefined;
/** @ignore */
CardStylePlatformBulkCreateOperationExperimentalRelease.attributeTypeMap = [
    {
        name: 'cardTheme',
        baseName: 'cardTheme',
        type: 'string',
    },
];
