// @mirohq/miro-api/dist/model/cardData.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardData = void 0;
/**
 * @internal
 * Contains card item data, such as the title, description, due date, or assignee ID.
 */
class CardData {
    constructor() {
        /**
         * A short text header for the card.
         */
        this['title'] = 'sample card item';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return CardData.attributeTypeMap;
    }
}
exports.CardData = CardData;
/** @ignore */
CardData.discriminator = undefined;
/** @ignore */
CardData.attributeTypeMap = [
    {
        name: 'assigneeId',
        baseName: 'assigneeId',
        type: 'string',
    },
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'dueDate',
        baseName: 'dueDate',
        type: 'Date',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
];
