// @mirohq/miro-api/dist/model/cardCreateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardCreateRequest = void 0;
class CardCreateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return CardCreateRequest.attributeTypeMap;
    }
}
exports.CardCreateRequest = CardCreateRequest;
/** @ignore */
CardCreateRequest.discriminator = undefined;
/** @ignore */
CardCreateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'CardData',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'CardStyle',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'Geometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
