// @mirohq/miro-api/dist/model/connectorChangesData.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectorChangesData = void 0;
/**
 * @internal
 * If both are provided, startItem.id must be different from endItem.id
 */
class ConnectorChangesData {
    /** @ignore */
    static getAttributeTypeMap() {
        return ConnectorChangesData.attributeTypeMap;
    }
}
exports.ConnectorChangesData = ConnectorChangesData;
/** @ignore */
ConnectorChangesData.discriminator = undefined;
/** @ignore */
ConnectorChangesData.attributeTypeMap = [
    {
        name: 'startItem',
        baseName: 'startItem',
        type: 'ItemConnectionChangesData',
    },
    {
        name: 'endItem',
        baseName: 'endItem',
        type: 'ItemConnectionChangesData',
    },
    {
        name: 'shape',
        baseName: 'shape',
        type: 'ConnectorChangesData.ShapeEnum',
    },
    {
        name: 'captions',
        baseName: 'captions',
        type: 'Array<Caption>',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'ConnectorStyle',
    },
];
(function (ConnectorChangesData) {
    ConnectorChangesData.ShapeEnum = {
        Straight: 'straight',
        Elbowed: 'elbowed',
        Curved: 'curved',
    };
})(ConnectorChangesData = exports.ConnectorChangesData || (exports.ConnectorChangesData = {}));
