// @mirohq/miro-api/dist/model/connectorCreationData.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectorCreationData = void 0;
/**
 * @internal
 * startItem.id must be different from endItem.id
 */
class ConnectorCreationData {
    /** @ignore */
    static getAttributeTypeMap() {
        return ConnectorCreationData.attributeTypeMap;
    }
}
exports.ConnectorCreationData = ConnectorCreationData;
/** @ignore */
ConnectorCreationData.discriminator = undefined;
/** @ignore */
ConnectorCreationData.attributeTypeMap = [
    {
        name: 'startItem',
        baseName: 'startItem',
        type: 'ItemConnectionCreationData',
    },
    {
        name: 'endItem',
        baseName: 'endItem',
        type: 'ItemConnectionCreationData',
    },
    {
        name: 'shape',
        baseName: 'shape',
        type: 'ConnectorCreationData.ShapeEnum',
    },
    {
        name: 'captions',
        baseName: 'captions',
        type: 'Array<Caption>',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'ConnectorStyle',
    },
];
(function (ConnectorCreationData) {
    ConnectorCreationData.ShapeEnum = {
        Straight: 'straight',
        Elbowed: 'elbowed',
        Curved: 'curved',
    };
})(ConnectorCreationData = exports.ConnectorCreationData || (exports.ConnectorCreationData = {}));
