// @mirohq/miro-api/dist/model/connectorWithLinks.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectorWithLinks = void 0;
/**
 * @internal
 * Contains the result data.
 */
class ConnectorWithLinks {
    constructor() {
        /**
         * The path type of the connector line, defines curvature. Default: curved.
         */
        this['shape'] = ConnectorWithLinks.ShapeEnum.Curved;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return ConnectorWithLinks.attributeTypeMap;
    }
}
exports.ConnectorWithLinks = ConnectorWithLinks;
/** @ignore */
ConnectorWithLinks.discriminator = undefined;
/** @ignore */
ConnectorWithLinks.attributeTypeMap = [
    {
        name: 'captions',
        baseName: 'captions',
        type: 'Array<Caption>',
    },
    {
        name: 'createdAt',
        baseName: 'createdAt',
        type: 'Date',
    },
    {
        name: 'createdBy',
        baseName: 'createdBy',
        type: 'CreatedBy',
    },
    {
        name: 'endItem',
        baseName: 'endItem',
        type: 'ItemConnectionWithLinks',
    },
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'isSupported',
        baseName: 'isSupported',
        type: 'boolean',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'SelfLink',
    },
    {
        name: 'modifiedAt',
        baseName: 'modifiedAt',
        type: 'Date',
    },
    {
        name: 'modifiedBy',
        baseName: 'modifiedBy',
        type: 'ModifiedBy',
    },
    {
        name: 'shape',
        baseName: 'shape',
        type: 'ConnectorWithLinks.ShapeEnum',
    },
    {
        name: 'startItem',
        baseName: 'startItem',
        type: 'ItemConnectionWithLinks',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'ConnectorStyle',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
(function (ConnectorWithLinks) {
    ConnectorWithLinks.ShapeEnum = {
        Straight: 'straight',
        Elbowed: 'elbowed',
        Curved: 'curved',
    };
})(ConnectorWithLinks = exports.ConnectorWithLinks || (exports.ConnectorWithLinks = {}));
