// @mirohq/miro-api/dist/model/connectorsCursorPaged.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectorsCursorPaged = void 0;
class ConnectorsCursorPaged {
    /** @ignore */
    static getAttributeTypeMap() {
        return ConnectorsCursorPaged.attributeTypeMap;
    }
}
exports.ConnectorsCursorPaged = ConnectorsCursorPaged;
/** @ignore */
ConnectorsCursorPaged.discriminator = undefined;
/** @ignore */
ConnectorsCursorPaged.attributeTypeMap = [
    {
        name: 'cursor',
        baseName: 'cursor',
        type: 'string',
    },
    {
        name: 'data',
        baseName: 'data',
        type: 'Array<ConnectorWithLinks>',
    },
    {
        name: 'limit',
        baseName: 'limit',
        type: 'number',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'PageLinks',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'total',
        baseName: 'total',
        type: 'number',
    },
];
