// @mirohq/miro-api/dist/model/copyBoardChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CopyBoardChanges = void 0;
class CopyBoardChanges {
    constructor() {
        /**
         * Name for the board.
         */
        this['name'] = 'Untitled';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return CopyBoardChanges.attributeTypeMap;
    }
}
exports.CopyBoardChanges = CopyBoardChanges;
/** @ignore */
CopyBoardChanges.discriminator = undefined;
/** @ignore */
CopyBoardChanges.attributeTypeMap = [
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'name',
        baseName: 'name',
        type: 'string',
    },
    {
        name: 'policy',
        baseName: 'policy',
        type: 'BoardPolicyChange',
    },
    {
        name: 'teamId',
        baseName: 'teamId',
        type: 'string',
    },
];
