// @mirohq/miro-api/dist/model/createBoardSubscriptionRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateBoardSubscriptionRequest = void 0;
/**
 * @internal
 * Contains the board ID associated with the webhook subscription, the webhook callback URL, and the status of the webhook subscription.
 */
class CreateBoardSubscriptionRequest {
    constructor() {
        /**
         * Indicates whether the status of the webhook subscription.`enabled`: Miro sends a webhook when an event occurs in the associated board. `disabled`: Miro does not send a webhook even when an event occurs in the associated board. `lost_access`: The user with which the webhook subscription is associated has lost access to the board. The user needs to regain access to the board, and then reenable the webhook subscription by updating the webhook subscription status to `enabled` by using the update webhook endpoint.
         */
        this['status'] = CreateBoardSubscriptionRequest.StatusEnum.Enabled;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return CreateBoardSubscriptionRequest.attributeTypeMap;
    }
}
exports.CreateBoardSubscriptionRequest = CreateBoardSubscriptionRequest;
/** @ignore */
CreateBoardSubscriptionRequest.discriminator = undefined;
/** @ignore */
CreateBoardSubscriptionRequest.attributeTypeMap = [
    {
        name: 'boardId',
        baseName: 'boardId',
        type: 'string',
    },
    {
        name: 'callbackUrl',
        baseName: 'callbackUrl',
        type: 'string',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'CreateBoardSubscriptionRequest.StatusEnum',
    },
];
(function (CreateBoardSubscriptionRequest) {
    CreateBoardSubscriptionRequest.StatusEnum = {
        Enabled: 'enabled',
        Disabled: 'disabled',
    };
})(CreateBoardSubscriptionRequest = exports.CreateBoardSubscriptionRequest || (exports.CreateBoardSubscriptionRequest = {}));
