// @mirohq/miro-api/dist/model/customField.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomField = void 0;
/**
 * @internal
 * Array where each object represents a custom preview field. Preview fields are displayed on the bottom half of the app card in the compact view.
 */
class CustomField {
    constructor() {
        /**
         * The shape of the icon on the preview field.
         */
        this['iconShape'] = CustomField.IconShapeEnum.Round;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return CustomField.attributeTypeMap;
    }
}
exports.CustomField = CustomField;
/** @ignore */
CustomField.discriminator = undefined;
/** @ignore */
CustomField.attributeTypeMap = [
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'string',
    },
    {
        name: 'iconShape',
        baseName: 'iconShape',
        type: 'CustomField.IconShapeEnum',
    },
    {
        name: 'iconUrl',
        baseName: 'iconUrl',
        type: 'string',
    },
    {
        name: 'textColor',
        baseName: 'textColor',
        type: 'string',
    },
    {
        name: 'tooltip',
        baseName: 'tooltip',
        type: 'string',
    },
    {
        name: 'value',
        baseName: 'value',
        type: 'string',
    },
];
(function (CustomField) {
    CustomField.IconShapeEnum = {
        Round: 'round',
        Square: 'square',
    };
})(CustomField = exports.CustomField || (exports.CustomField = {}));
