// @mirohq/miro-api/dist/model/customFieldPlatformTags.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldPlatformTags = void 0;
/**
 * @internal
 * Array where each object represents a custom preview field. Preview fields are displayed on the bottom half of the app card in the compact view.
 */
class CustomFieldPlatformTags {
    constructor() {
        /**
         * The shape of the icon on the preview field.
         */
        this['iconShape'] = CustomFieldPlatformTags.IconShapeEnum.Round;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return CustomFieldPlatformTags.attributeTypeMap;
    }
}
exports.CustomFieldPlatformTags = CustomFieldPlatformTags;
/** @ignore */
CustomFieldPlatformTags.discriminator = undefined;
/** @ignore */
CustomFieldPlatformTags.attributeTypeMap = [
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'string',
    },
    {
        name: 'iconShape',
        baseName: 'iconShape',
        type: 'CustomFieldPlatformTags.IconShapeEnum',
    },
    {
        name: 'iconUrl',
        baseName: 'iconUrl',
        type: 'string',
    },
    {
        name: 'textColor',
        baseName: 'textColor',
        type: 'string',
    },
    {
        name: 'tooltip',
        baseName: 'tooltip',
        type: 'string',
    },
    {
        name: 'value',
        baseName: 'value',
        type: 'string',
    },
];
(function (CustomFieldPlatformTags) {
    CustomFieldPlatformTags.IconShapeEnum = {
        Round: 'round',
        Square: 'square',
    };
})(CustomFieldPlatformTags = exports.CustomFieldPlatformTags || (exports.CustomFieldPlatformTags = {}));
