// @mirohq/miro-api/dist/model/customFieldPlatformBulkCreateOperationExperimentalRelease.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldPlatformBulkCreateOperationExperimentalRelease = void 0;
/**
 * @internal
 * Array where each object represents a custom preview field. Preview fields are displayed on the bottom half of the app card in the compact view.
 */
class CustomFieldPlatformBulkCreateOperationExperimentalRelease {
    constructor() {
        /**
         * The shape of the icon on the preview field.
         */
        this['iconShape'] = CustomFieldPlatformBulkCreateOperationExperimentalRelease.IconShapeEnum.Round;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return CustomFieldPlatformBulkCreateOperationExperimentalRelease.attributeTypeMap;
    }
}
exports.CustomFieldPlatformBulkCreateOperationExperimentalRelease = CustomFieldPlatformBulkCreateOperationExperimentalRelease;
/** @ignore */
CustomFieldPlatformBulkCreateOperationExperimentalRelease.discriminator = undefined;
/** @ignore */
CustomFieldPlatformBulkCreateOperationExperimentalRelease.attributeTypeMap = [
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'string',
    },
    {
        name: 'iconShape',
        baseName: 'iconShape',
        type: 'CustomFieldPlatformBulkCreateOperationExperimentalRelease.IconShapeEnum',
    },
    {
        name: 'iconUrl',
        baseName: 'iconUrl',
        type: 'string',
    },
    {
        name: 'textColor',
        baseName: 'textColor',
        type: 'string',
    },
    {
        name: 'tooltip',
        baseName: 'tooltip',
        type: 'string',
    },
    {
        name: 'value',
        baseName: 'value',
        type: 'string',
    },
];
(function (CustomFieldPlatformBulkCreateOperationExperimentalRelease) {
    CustomFieldPlatformBulkCreateOperationExperimentalRelease.IconShapeEnum = {
        Round: 'round',
        Square: 'square',
    };
})(CustomFieldPlatformBulkCreateOperationExperimentalRelease = exports.CustomFieldPlatformBulkCreateOperationExperimentalRelease || (exports.CustomFieldPlatformBulkCreateOperationExperimentalRelease = {}));
