// @mirohq/miro-api/dist/model/dataClassificationLabel.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataClassificationLabel = void 0;
/**
 * @internal
 * Data classification label
 */
class DataClassificationLabel {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'data-classification-label';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return DataClassificationLabel.attributeTypeMap;
    }
}
exports.DataClassificationLabel = DataClassificationLabel;
/** @ignore */
DataClassificationLabel.discriminator = undefined;
/** @ignore */
DataClassificationLabel.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'color',
        baseName: 'color',
        type: 'string',
    },
    {
        name: '_default',
        baseName: 'default',
        type: 'boolean',
    },
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'name',
        baseName: 'name',
        type: 'string',
    },
    {
        name: 'orderNumber',
        baseName: 'orderNumber',
        type: 'number',
    },
    {
        name: 'sharingRecommendation',
        baseName: 'sharingRecommendation',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
