// @mirohq/miro-api/dist/model/dataClassificationTeamSettings.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataClassificationTeamSettings = void 0;
class DataClassificationTeamSettings {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'team-data-classification-settings';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return DataClassificationTeamSettings.attributeTypeMap;
    }
}
exports.DataClassificationTeamSettings = DataClassificationTeamSettings;
/** @ignore */
DataClassificationTeamSettings.discriminator = undefined;
/** @ignore */
DataClassificationTeamSettings.attributeTypeMap = [
    {
        name: 'defaultLabelId',
        baseName: 'defaultLabelId',
        type: 'string',
    },
    {
        name: 'enabled',
        baseName: 'enabled',
        type: 'boolean',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
