// @mirohq/miro-api/dist/model/dataClassificationOrganizationSettings.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataClassificationOrganizationSettings = void 0;
class DataClassificationOrganizationSettings {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'data-classification-organization-settings';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return DataClassificationOrganizationSettings.attributeTypeMap;
    }
}
exports.DataClassificationOrganizationSettings = DataClassificationOrganizationSettings;
/** @ignore */
DataClassificationOrganizationSettings.discriminator = undefined;
/** @ignore */
DataClassificationOrganizationSettings.attributeTypeMap = [
    {
        name: 'enabled',
        baseName: 'enabled',
        type: 'boolean',
    },
    {
        name: 'labels',
        baseName: 'labels',
        type: 'Array<DataClassificationLabel>',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
