// @mirohq/miro-api/dist/model/documentUrlDataPlatformBulkCreateOperationExperimentalRelease.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentUrlDataPlatformBulkCreateOperationExperimentalRelease = void 0;
/**
 * @internal
 * Contains information about the document URL.
 */
class DocumentUrlDataPlatformBulkCreateOperationExperimentalRelease {
    constructor() {
        /**
         * URL where the document is hosted.
         */
        this['url'] = 'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return DocumentUrlDataPlatformBulkCreateOperationExperimentalRelease.attributeTypeMap;
    }
}
exports.DocumentUrlDataPlatformBulkCreateOperationExperimentalRelease = DocumentUrlDataPlatformBulkCreateOperationExperimentalRelease;
/** @ignore */
DocumentUrlDataPlatformBulkCreateOperationExperimentalRelease.discriminator = undefined;
/** @ignore */
DocumentUrlDataPlatformBulkCreateOperationExperimentalRelease.attributeTypeMap = [
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'url',
        baseName: 'url',
        type: 'string',
    },
];
