// @mirohq/miro-api/dist/model/documentCreateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentCreateRequest = void 0;
class DocumentCreateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return DocumentCreateRequest.attributeTypeMap;
    }
}
exports.DocumentCreateRequest = DocumentCreateRequest;
/** @ignore */
DocumentCreateRequest.discriminator = undefined;
/** @ignore */
DocumentCreateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'DocumentUrlData',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'FixedRatioGeometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
