// @mirohq/miro-api/dist/model/documentUpdateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentUpdateRequest = void 0;
class DocumentUpdateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return DocumentUpdateRequest.attributeTypeMap;
    }
}
exports.DocumentUpdateRequest = DocumentUpdateRequest;
/** @ignore */
DocumentUpdateRequest.discriminator = undefined;
/** @ignore */
DocumentUpdateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'DocumentUrlDataChanges',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'FixedRatioGeometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
