// @mirohq/miro-api/dist/model/embedCreateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbedCreateRequest = void 0;
class EmbedCreateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return EmbedCreateRequest.attributeTypeMap;
    }
}
exports.EmbedCreateRequest = EmbedCreateRequest;
/** @ignore */
EmbedCreateRequest.discriminator = undefined;
/** @ignore */
EmbedCreateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'EmbedUrlData',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'FixedRatioNoRotationGeometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
