// @mirohq/miro-api/dist/model/embedUrlDataChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbedUrlDataChanges = void 0;
/**
 * @internal
 * Contains information about the embed URL.
 */
class EmbedUrlDataChanges {
    /** @ignore */
    static getAttributeTypeMap() {
        return EmbedUrlDataChanges.attributeTypeMap;
    }
}
exports.EmbedUrlDataChanges = EmbedUrlDataChanges;
/** @ignore */
EmbedUrlDataChanges.discriminator = undefined;
/** @ignore */
EmbedUrlDataChanges.attributeTypeMap = [
    {
        name: 'mode',
        baseName: 'mode',
        type: 'EmbedUrlDataChanges.ModeEnum',
    },
    {
        name: 'previewUrl',
        baseName: 'previewUrl',
        type: 'string',
    },
    {
        name: 'url',
        baseName: 'url',
        type: 'string',
    },
];
(function (EmbedUrlDataChanges) {
    EmbedUrlDataChanges.ModeEnum = {
        Inline: 'inline',
        Modal: 'modal',
    };
})(EmbedUrlDataChanges = exports.EmbedUrlDataChanges || (exports.EmbedUrlDataChanges = {}));
