// @mirohq/miro-api/dist/model/embedDataResponse.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbedDataResponse = void 0;
class EmbedDataResponse {
    /** @ignore */
    static getAttributeTypeMap() {
        return EmbedDataResponse.attributeTypeMap;
    }
}
exports.EmbedDataResponse = EmbedDataResponse;
/** @ignore */
EmbedDataResponse.discriminator = undefined;
/** @ignore */
EmbedDataResponse.attributeTypeMap = [
    {
        name: 'contentType',
        baseName: 'contentType',
        type: 'string',
    },
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'html',
        baseName: 'html',
        type: 'string',
    },
    {
        name: 'mode',
        baseName: 'mode',
        type: 'EmbedDataResponse.ModeEnum',
    },
    {
        name: 'previewUrl',
        baseName: 'previewUrl',
        type: 'string',
    },
    {
        name: 'providerName',
        baseName: 'providerName',
        type: 'string',
    },
    {
        name: 'providerUrl',
        baseName: 'providerUrl',
        type: 'string',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'url',
        baseName: 'url',
        type: 'string',
    },
];
(function (EmbedDataResponse) {
    EmbedDataResponse.ModeEnum = {
        Inline: 'inline',
        Modal: 'modal',
    };
})(EmbedDataResponse = exports.EmbedDataResponse || (exports.EmbedDataResponse = {}));
