// @mirohq/miro-api/dist/model/embedUpdateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbedUpdateRequest = void 0;
class EmbedUpdateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return EmbedUpdateRequest.attributeTypeMap;
    }
}
exports.EmbedUpdateRequest = EmbedUpdateRequest;
/** @ignore */
EmbedUpdateRequest.discriminator = undefined;
/** @ignore */
EmbedUpdateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'EmbedUrlDataChanges',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'FixedRatioNoRotationGeometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
