// @mirohq/miro-api/dist/model/embedData.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbedData = void 0;
class EmbedData {
    /** @ignore */
    static getAttributeTypeMap() {
        return EmbedData.attributeTypeMap;
    }
}
exports.EmbedData = EmbedData;
/** @ignore */
EmbedData.discriminator = undefined;
/** @ignore */
EmbedData.attributeTypeMap = [
    {
        name: 'contentType',
        baseName: 'contentType',
        type: 'string',
    },
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'html',
        baseName: 'html',
        type: 'string',
    },
    {
        name: 'mode',
        baseName: 'mode',
        type: 'EmbedData.ModeEnum',
    },
    {
        name: 'previewUrl',
        baseName: 'previewUrl',
        type: 'string',
    },
    {
        name: 'providerName',
        baseName: 'providerName',
        type: 'string',
    },
    {
        name: 'providerUrl',
        baseName: 'providerUrl',
        type: 'string',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'url',
        baseName: 'url',
        type: 'string',
    },
];
(function (EmbedData) {
    EmbedData.ModeEnum = {
        Inline: 'inline',
        Modal: 'modal',
    };
})(EmbedData = exports.EmbedData || (exports.EmbedData = {}));
