// @mirohq/miro-api/dist/model/embedItem.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbedItem = void 0;
class EmbedItem {
    /** @ignore */
    static getAttributeTypeMap() {
        return EmbedItem.attributeTypeMap;
    }
}
exports.EmbedItem = EmbedItem;
/** @ignore */
EmbedItem.discriminator = undefined;
/** @ignore */
EmbedItem.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'data',
        baseName: 'data',
        type: 'EmbedData',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'Position',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'Geometry',
    },
    {
        name: 'createdAt',
        baseName: 'createdAt',
        type: 'Date',
    },
    {
        name: 'createdBy',
        baseName: 'createdBy',
        type: 'CreatedBy',
    },
    {
        name: 'modifiedAt',
        baseName: 'modifiedAt',
        type: 'Date',
    },
    {
        name: 'modifiedBy',
        baseName: 'modifiedBy',
        type: 'ModifiedBy',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'ParentLinksEnvelope',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'WidgetLinks',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
