// @mirohq/miro-api/dist/model/embedUrlDataPlatformBulkCreateOperationExperimentalRelease.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease = void 0;
/**
 * @internal
 * Contains information about the embed URL.
 */
class EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease {
    constructor() {
        /**
         * A [valid URL](https://developers.miro.com/reference/data#embeddata) pointing to the content resource that you want to embed in the board. Possible transport protocols: HTTP, HTTPS.
         */
        this['url'] = 'https://www.youtube.com/watch?v=HlVSNEiFCBk';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease.attributeTypeMap;
    }
}
exports.EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease = EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease;
/** @ignore */
EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease.discriminator = undefined;
/** @ignore */
EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease.attributeTypeMap = [
    {
        name: 'mode',
        baseName: 'mode',
        type: 'EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease.ModeEnum',
    },
    {
        name: 'previewUrl',
        baseName: 'previewUrl',
        type: 'string',
    },
    {
        name: 'url',
        baseName: 'url',
        type: 'string',
    },
];
(function (EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease) {
    EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease.ModeEnum = {
        Inline: 'inline',
        Modal: 'modal',
    };
})(EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease = exports.EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease || (exports.EmbedUrlDataPlatformBulkCreateOperationExperimentalRelease = {}));
