// @mirohq/miro-api/dist/model/enterpriseGetOrganizationMembers200Response.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnterpriseGetOrganizationMembers200Response = void 0;
class EnterpriseGetOrganizationMembers200Response {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'cursor-list';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return EnterpriseGetOrganizationMembers200Response.attributeTypeMap;
    }
}
exports.EnterpriseGetOrganizationMembers200Response = EnterpriseGetOrganizationMembers200Response;
/** @ignore */
EnterpriseGetOrganizationMembers200Response.discriminator = undefined;
/** @ignore */
EnterpriseGetOrganizationMembers200Response.attributeTypeMap = [
    {
        name: 'limit',
        baseName: 'limit',
        type: 'number',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'data',
        baseName: 'data',
        type: 'Array<OrganizationMember>',
    },
    {
        name: 'cursor',
        baseName: 'cursor',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
