// @mirohq/miro-api/dist/model/error404.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Error404 = void 0;
class Error404 {
    constructor() {
        /**
         * HTTP status code.
         */
        this['status'] = 404;
        this['type'] = 'error';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return Error404.attributeTypeMap;
    }
}
exports.Error404 = Error404;
/** @ignore */
Error404.discriminator = undefined;
/** @ignore */
Error404.attributeTypeMap = [
    {
        name: 'status',
        baseName: 'status',
        type: 'number',
    },
    {
        name: 'code',
        baseName: 'code',
        type: 'string',
    },
    {
        name: 'message',
        baseName: 'message',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
