// @mirohq/miro-api/dist/model/error400.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Error400 = void 0;
class Error400 {
    constructor() {
        /**
         * HTTP status code.
         */
        this['status'] = 400;
        /**
         * Description of the status code.
         */
        this['code'] = 'invalidParameters';
        this['type'] = 'error';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return Error400.attributeTypeMap;
    }
}
exports.Error400 = Error400;
/** @ignore */
Error400.discriminator = undefined;
/** @ignore */
Error400.attributeTypeMap = [
    {
        name: 'status',
        baseName: 'status',
        type: 'number',
    },
    {
        name: 'code',
        baseName: 'code',
        type: 'string',
    },
    {
        name: 'message',
        baseName: 'message',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
