// @mirohq/miro-api/dist/model/fixedRatioGeometry.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FixedRatioGeometry = void 0;
/**
 * @internal
 * Contains geometrical information about the item, such as its width or rotation. You can set either the width or height, you cannot set both the width and height at the same time.
 */
class FixedRatioGeometry {
    /** @ignore */
    static getAttributeTypeMap() {
        return FixedRatioGeometry.attributeTypeMap;
    }
}
exports.FixedRatioGeometry = FixedRatioGeometry;
/** @ignore */
FixedRatioGeometry.discriminator = undefined;
/** @ignore */
FixedRatioGeometry.attributeTypeMap = [
    {
        name: 'height',
        baseName: 'height',
        type: 'number',
    },
    {
        name: 'width',
        baseName: 'width',
        type: 'number',
    },
    {
        name: 'rotation',
        baseName: 'rotation',
        type: 'number',
    },
];
