// @mirohq/miro-api/dist/model/fixedRatioNoRotationGeometry.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FixedRatioNoRotationGeometry = void 0;
/**
 * @internal
 * Contains geometrical information about the item. You can set either the width or height. You cannot set both the width and height at the same time.
 */
class FixedRatioNoRotationGeometry {
    /** @ignore */
    static getAttributeTypeMap() {
        return FixedRatioNoRotationGeometry.attributeTypeMap;
    }
}
exports.FixedRatioNoRotationGeometry = FixedRatioNoRotationGeometry;
/** @ignore */
FixedRatioNoRotationGeometry.discriminator = undefined;
/** @ignore */
FixedRatioNoRotationGeometry.attributeTypeMap = [
    {
        name: 'height',
        baseName: 'height',
        type: 'number',
    },
    {
        name: 'width',
        baseName: 'width',
        type: 'number',
    },
];
