// @mirohq/miro-api/dist/model/frameDataPlatformTags.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrameDataPlatformTags = void 0;
/**
 * @internal
 * Contains frame item data, such as the title, frame type, or frame format.
 */
class FrameDataPlatformTags {
    constructor() {
        /**
         * Only custom frames are supported at the moment.
         */
        this['format'] = FrameDataPlatformTags.FormatEnum.Custom;
        /**
         * Only free form frames are supported at the moment.
         */
        this['type'] = FrameDataPlatformTags.TypeEnum.Freeform;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return FrameDataPlatformTags.attributeTypeMap;
    }
}
exports.FrameDataPlatformTags = FrameDataPlatformTags;
/** @ignore */
FrameDataPlatformTags.discriminator = undefined;
/** @ignore */
FrameDataPlatformTags.attributeTypeMap = [
    {
        name: 'format',
        baseName: 'format',
        type: 'FrameDataPlatformTags.FormatEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'FrameDataPlatformTags.TypeEnum',
    },
];
(function (FrameDataPlatformTags) {
    FrameDataPlatformTags.FormatEnum = {
        Custom: 'custom',
        Desktop: 'desktop',
        Phone: 'phone',
        Tablet: 'tablet',
        A4: 'a4',
        Letter: 'letter',
        Ratio1x1: 'ratio_1x1',
        Ratio4x3: 'ratio_4x3',
        Ratio16x9: 'ratio_16x9',
    };
    FrameDataPlatformTags.TypeEnum = {
        Freeform: 'freeform',
        Heap: 'heap',
        Grid: 'grid',
        Rows: 'rows',
        Columns: 'columns',
    };
})(FrameDataPlatformTags = exports.FrameDataPlatformTags || (exports.FrameDataPlatformTags = {}));
