// @mirohq/miro-api/dist/model/frameChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrameChanges = void 0;
/**
 * @internal
 * Contains frame item data, such as the title, frame type, or frame format.
 */
class FrameChanges {
    constructor() {
        /**
         * Only custom frames are supported at the moment.
         */
        this['format'] = FrameChanges.FormatEnum.Custom;
        /**
         * Title of the frame. This title appears at the top of the frame.
         */
        this['title'] = 'Sample frame title';
        /**
         * Only free form frames are supported at the moment.
         */
        this['type'] = FrameChanges.TypeEnum.Freeform;
        /**
         * Hide or reveal the content inside a frame (Enterprise plan only).
         */
        this['showContent'] = true;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return FrameChanges.attributeTypeMap;
    }
}
exports.FrameChanges = FrameChanges;
/** @ignore */
FrameChanges.discriminator = undefined;
/** @ignore */
FrameChanges.attributeTypeMap = [
    {
        name: 'format',
        baseName: 'format',
        type: 'FrameChanges.FormatEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'FrameChanges.TypeEnum',
    },
    {
        name: 'showContent',
        baseName: 'showContent',
        type: 'boolean',
    },
];
(function (FrameChanges) {
    FrameChanges.FormatEnum = {
        Custom: 'custom',
    };
    FrameChanges.TypeEnum = {
        Freeform: 'freeform',
    };
})(FrameChanges = exports.FrameChanges || (exports.FrameChanges = {}));
