// @mirohq/miro-api/dist/model/frameData.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrameData = void 0;
/**
 * @internal
 * Contains frame item data, such as the title, frame type, or frame format.
 */
class FrameData {
    constructor() {
        /**
         * Only custom frames are supported at the moment.
         */
        this['format'] = FrameData.FormatEnum.Custom;
        /**
         * Only free form frames are supported at the moment.
         */
        this['type'] = FrameData.TypeEnum.Freeform;
        /**
         * Hide or reveal the content inside a frame (Enterprise plan only).
         */
        this['showContent'] = true;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return FrameData.attributeTypeMap;
    }
}
exports.FrameData = FrameData;
/** @ignore */
FrameData.discriminator = undefined;
/** @ignore */
FrameData.attributeTypeMap = [
    {
        name: 'format',
        baseName: 'format',
        type: 'FrameData.FormatEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'FrameData.TypeEnum',
    },
    {
        name: 'showContent',
        baseName: 'showContent',
        type: 'boolean',
    },
];
(function (FrameData) {
    FrameData.FormatEnum = {
        Custom: 'custom',
        Desktop: 'desktop',
        Phone: 'phone',
        Tablet: 'tablet',
        A4: 'a4',
        Letter: 'letter',
        Ratio1x1: 'ratio_1x1',
        Ratio4x3: 'ratio_4x3',
        Ratio16x9: 'ratio_16x9',
    };
    FrameData.TypeEnum = {
        Freeform: 'freeform',
        Heap: 'heap',
        Grid: 'grid',
        Rows: 'rows',
        Columns: 'columns',
    };
})(FrameData = exports.FrameData || (exports.FrameData = {}));
