// @mirohq/miro-api/dist/model/frameCreateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrameCreateRequest = void 0;
class FrameCreateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return FrameCreateRequest.attributeTypeMap;
    }
}
exports.FrameCreateRequest = FrameCreateRequest;
/** @ignore */
FrameCreateRequest.discriminator = undefined;
/** @ignore */
FrameCreateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'FrameChanges',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'FrameStyle',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'GeometryNoRotation',
    },
];
