// @mirohq/miro-api/dist/model/genericSubscriptionsCursorPaged.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenericSubscriptionsCursorPaged = void 0;
class GenericSubscriptionsCursorPaged {
    /** @ignore */
    static getAttributeTypeMap() {
        return GenericSubscriptionsCursorPaged.attributeTypeMap;
    }
}
exports.GenericSubscriptionsCursorPaged = GenericSubscriptionsCursorPaged;
/** @ignore */
GenericSubscriptionsCursorPaged.discriminator = undefined;
/** @ignore */
GenericSubscriptionsCursorPaged.attributeTypeMap = [
    {
        name: 'cursor',
        baseName: 'cursor',
        type: 'string',
    },
    {
        name: 'data',
        baseName: 'data',
        type: 'Array<GenericSubscription>',
    },
    {
        name: 'limit',
        baseName: 'limit',
        type: 'number',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'total',
        baseName: 'total',
        type: 'number',
    },
];
