// @mirohq/miro-api/dist/model/genericItemCursorPaged.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenericItemCursorPaged = void 0;
class GenericItemCursorPaged {
    /** @ignore */
    static getAttributeTypeMap() {
        return GenericItemCursorPaged.attributeTypeMap;
    }
}
exports.GenericItemCursorPaged = GenericItemCursorPaged;
/** @ignore */
GenericItemCursorPaged.discriminator = undefined;
/** @ignore */
GenericItemCursorPaged.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'Array<GenericItem>',
    },
    {
        name: 'total',
        baseName: 'total',
        type: 'number',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'cursor',
        baseName: 'cursor',
        type: 'string',
    },
    {
        name: 'limit',
        baseName: 'limit',
        type: 'number',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'PageLinks',
    },
];
