// @mirohq/miro-api/dist/model/geometryPlatformContainers.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeometryPlatformContainers = void 0;
/**
 * @internal
 * Contains geometrical information about the item, such as its width or height.
 */
class GeometryPlatformContainers {
    /** @ignore */
    static getAttributeTypeMap() {
        return GeometryPlatformContainers.attributeTypeMap;
    }
}
exports.GeometryPlatformContainers = GeometryPlatformContainers;
/** @ignore */
GeometryPlatformContainers.discriminator = undefined;
/** @ignore */
GeometryPlatformContainers.attributeTypeMap = [
    {
        name: 'height',
        baseName: 'height',
        type: 'number',
    },
    {
        name: 'rotation',
        baseName: 'rotation',
        type: 'number',
    },
    {
        name: 'width',
        baseName: 'width',
        type: 'number',
    },
];
