// @mirohq/miro-api/dist/model/getTagsFromItem400Response.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTagsFromItem400Response = void 0;
class GetTagsFromItem400Response {
    /** @ignore */
    static getAttributeTypeMap() {
        return GetTagsFromItem400Response.attributeTypeMap;
    }
}
exports.GetTagsFromItem400Response = GetTagsFromItem400Response;
/** @ignore */
GetTagsFromItem400Response.discriminator = undefined;
/** @ignore */
GetTagsFromItem400Response.attributeTypeMap = [
    {
        name: 'code',
        baseName: 'code',
        type: 'string',
    },
    {
        name: 'message',
        baseName: 'message',
        type: 'string',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'number',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
