// @mirohq/miro-api/dist/model/imageUpdateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageUpdateRequest = void 0;
class ImageUpdateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return ImageUpdateRequest.attributeTypeMap;
    }
}
exports.ImageUpdateRequest = ImageUpdateRequest;
/** @ignore */
ImageUpdateRequest.discriminator = undefined;
/** @ignore */
ImageUpdateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'ImageUrlDataChanges',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'FixedRatioGeometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
