// @mirohq/miro-api/dist/model/imageUrlData.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageUrlData = void 0;
/**
 * @internal
 * Contains information about the image URL.
 */
class ImageUrlData {
    constructor() {
        /**
         * URL of the image.
         */
        this['url'] = 'https://miro.com/static/images/page/mr-index/localization/en/slider/ideation_brainstorming.png';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return ImageUrlData.attributeTypeMap;
    }
}
exports.ImageUrlData = ImageUrlData;
/** @ignore */
ImageUrlData.discriminator = undefined;
/** @ignore */
ImageUrlData.attributeTypeMap = [
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'url',
        baseName: 'url',
        type: 'string',
    },
];
