// @mirohq/miro-api/dist/model/imageUrlDataChangesPlatformBulkCreateOperationExperimentalRelease.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageUrlDataChangesPlatformBulkCreateOperationExperimentalRelease = void 0;
/**
 * @internal
 * Contains information about the image URL.
 */
class ImageUrlDataChangesPlatformBulkCreateOperationExperimentalRelease {
    /** @ignore */
    static getAttributeTypeMap() {
        return ImageUrlDataChangesPlatformBulkCreateOperationExperimentalRelease.attributeTypeMap;
    }
}
exports.ImageUrlDataChangesPlatformBulkCreateOperationExperimentalRelease = ImageUrlDataChangesPlatformBulkCreateOperationExperimentalRelease;
/** @ignore */
ImageUrlDataChangesPlatformBulkCreateOperationExperimentalRelease.discriminator = undefined;
/** @ignore */
ImageUrlDataChangesPlatformBulkCreateOperationExperimentalRelease.attributeTypeMap = [
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'url',
        baseName: 'url',
        type: 'string',
    },
];
