// @mirohq/miro-api/dist/model/itemTypeChange.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemTypeChange = void 0;
/**
 * @internal
 * Type of item that you want to create.
 */
var ItemTypeChange;
(function (ItemTypeChange) {
    ItemTypeChange[ItemTypeChange["AppCard"] = 'app_card'] = "AppCard";
    ItemTypeChange[ItemTypeChange["Text"] = 'text'] = "Text";
    ItemTypeChange[ItemTypeChange["Shape"] = 'shape'] = "Shape";
    ItemTypeChange[ItemTypeChange["StickyNote"] = 'sticky_note'] = "StickyNote";
    ItemTypeChange[ItemTypeChange["Image"] = 'image'] = "Image";
    ItemTypeChange[ItemTypeChange["Document"] = 'document'] = "Document";
    ItemTypeChange[ItemTypeChange["Card"] = 'card'] = "Card";
    ItemTypeChange[ItemTypeChange["Frame"] = 'frame'] = "Frame";
    ItemTypeChange[ItemTypeChange["Embed"] = 'embed'] = "Embed";
})(ItemTypeChange = exports.ItemTypeChange || (exports.ItemTypeChange = {}));
