// @mirohq/miro-api/dist/model/itemConnectionChangesData.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemConnectionChangesData = void 0;
/**
 * @internal
 * The ending point of the connector. If startItem is also provided, endItem.id must be different from startItem.id
 */
class ItemConnectionChangesData {
    /** @ignore */
    static getAttributeTypeMap() {
        return ItemConnectionChangesData.attributeTypeMap;
    }
}
exports.ItemConnectionChangesData = ItemConnectionChangesData;
/** @ignore */
ItemConnectionChangesData.discriminator = undefined;
/** @ignore */
ItemConnectionChangesData.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'RelativeOffset',
    },
    {
        name: 'snapTo',
        baseName: 'snapTo',
        type: 'ItemConnectionChangesData.SnapToEnum',
    },
];
(function (ItemConnectionChangesData) {
    ItemConnectionChangesData.SnapToEnum = {
        Auto: 'auto',
        Top: 'top',
        Right: 'right',
        Bottom: 'bottom',
        Left: 'left',
    };
})(ItemConnectionChangesData = exports.ItemConnectionChangesData || (exports.ItemConnectionChangesData = {}));
