// @mirohq/miro-api/dist/model/itemDataChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemDataChanges = void 0;
/**
 * @internal
 * Provides information about item-specific fields for update request.
 */
class ItemDataChanges {
    constructor() {
        /**
         * Status indicating whether an app card is connected and in sync with the source. When the source for the app card is deleted, the status returns `disabled`.
         */
        this['status'] = ItemDataChanges.StatusEnum.Disconnected;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return ItemDataChanges.attributeTypeMap;
    }
}
exports.ItemDataChanges = ItemDataChanges;
/** @ignore */
ItemDataChanges.discriminator = undefined;
/** @ignore */
ItemDataChanges.attributeTypeMap = [
    {
        name: 'description',
        baseName: 'description',
        type: 'string',
    },
    {
        name: 'fields',
        baseName: 'fields',
        type: 'Array<CustomField>',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'ItemDataChanges.StatusEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'url',
        baseName: 'url',
        type: 'string',
    },
];
(function (ItemDataChanges) {
    ItemDataChanges.StatusEnum = {
        Disconnected: 'disconnected',
        Connected: 'connected',
        Disabled: 'disabled',
    };
})(ItemDataChanges = exports.ItemDataChanges || (exports.ItemDataChanges = {}));
