// @mirohq/miro-api/dist/model/itemConnectionWithLinks.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemConnectionWithLinks = void 0;
/**
 * @internal
 * The starting point of the connector.
 */
class ItemConnectionWithLinks {
    /** @ignore */
    static getAttributeTypeMap() {
        return ItemConnectionWithLinks.attributeTypeMap;
    }
}
exports.ItemConnectionWithLinks = ItemConnectionWithLinks;
/** @ignore */
ItemConnectionWithLinks.discriminator = undefined;
/** @ignore */
ItemConnectionWithLinks.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'SelfLink',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'RelativeOffset',
    },
];
