// @mirohq/miro-api/dist/model/itemsPage.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemsPage = void 0;
/**
 * @internal
 * Contains cursor-based items page information.
 */
class ItemsPage {
    /** @ignore */
    static getAttributeTypeMap() {
        return ItemsPage.attributeTypeMap;
    }
}
exports.ItemsPage = ItemsPage;
/** @ignore */
ItemsPage.discriminator = undefined;
/** @ignore */
ItemsPage.attributeTypeMap = [
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'limit',
        baseName: 'limit',
        type: 'number',
    },
    {
        name: 'total',
        baseName: 'total',
        type: 'number',
    },
    {
        name: 'data',
        baseName: 'data',
        type: 'Array<Item>',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'PageLinks',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
