// @mirohq/miro-api/dist/model/itemConnectionCreationData.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemConnectionCreationData = void 0;
/**
 * @internal
 * The end point of the connector. endItem.id must be different from startItem.id
 */
class ItemConnectionCreationData {
    /** @ignore */
    static getAttributeTypeMap() {
        return ItemConnectionCreationData.attributeTypeMap;
    }
}
exports.ItemConnectionCreationData = ItemConnectionCreationData;
/** @ignore */
ItemConnectionCreationData.discriminator = undefined;
/** @ignore */
ItemConnectionCreationData.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'RelativeOffset',
    },
    {
        name: 'snapTo',
        baseName: 'snapTo',
        type: 'ItemConnectionCreationData.SnapToEnum',
    },
];
(function (ItemConnectionCreationData) {
    ItemConnectionCreationData.SnapToEnum = {
        Auto: 'auto',
        Top: 'top',
        Right: 'right',
        Bottom: 'bottom',
        Left: 'left',
    };
})(ItemConnectionCreationData = exports.ItemConnectionCreationData || (exports.ItemConnectionCreationData = {}));
