// @mirohq/miro-api/dist/model/itemChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemChanges = void 0;
/**
 * @internal
 * Updates one or more items in one request. You can update up to 20 items per request.
 */
class ItemChanges {
    /** @ignore */
    static getAttributeTypeMap() {
        return ItemChanges.attributeTypeMap;
    }
}
exports.ItemChanges = ItemChanges;
/** @ignore */
ItemChanges.discriminator = undefined;
/** @ignore */
ItemChanges.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'ItemTypeChange',
    },
    {
        name: 'data',
        baseName: 'data',
        type: 'ItemDataChanges',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'ItemStyle',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'Geometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
