// @mirohq/miro-api/dist/model/item.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Item = void 0;
/**
 * @internal
 * Contains information about an item.
 */
class Item {
    /** @ignore */
    static getAttributeTypeMap() {
        return Item.attributeTypeMap;
    }
}
exports.Item = Item;
/** @ignore */
Item.discriminator = undefined;
/** @ignore */
Item.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'data',
        baseName: 'data',
        type: 'ItemData',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'ItemStyle',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'Position',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'Geometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'ParentWithLinks',
    },
    {
        name: 'isSupported',
        baseName: 'isSupported',
        type: 'boolean',
    },
    {
        name: 'createdBy',
        baseName: 'createdBy',
        type: 'CreatedBy',
    },
    {
        name: 'createdAt',
        baseName: 'createdAt',
        type: 'Date',
    },
    {
        name: 'modifiedBy',
        baseName: 'modifiedBy',
        type: 'ModifiedBy',
    },
    {
        name: 'modifiedAt',
        baseName: 'modifiedAt',
        type: 'Date',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'SelfLink',
    },
];
