// @mirohq/miro-api/dist/model/itemCreate.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemCreate = void 0;
/**
 * @internal
 * Creates one or more items in one request. You can create up to 20 items per request.
 */
class ItemCreate {
    /** @ignore */
    static getAttributeTypeMap() {
        return ItemCreate.attributeTypeMap;
    }
}
exports.ItemCreate = ItemCreate;
/** @ignore */
ItemCreate.discriminator = undefined;
/** @ignore */
ItemCreate.attributeTypeMap = [
    {
        name: 'type',
        baseName: 'type',
        type: 'ItemTypeChange',
    },
    {
        name: 'data',
        baseName: 'data',
        type: 'ItemDataCreate',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'ItemStyle',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'Geometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
