// @mirohq/miro-api/dist/model/modelError.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelError = void 0;
/**
 * @internal
 * Error information
 */
class ModelError {
    /** @ignore */
    static getAttributeTypeMap() {
        return ModelError.attributeTypeMap;
    }
}
exports.ModelError = ModelError;
/** @ignore */
ModelError.discriminator = undefined;
/** @ignore */
ModelError.attributeTypeMap = [
    {
        name: 'code',
        baseName: 'code',
        type: 'string',
    },
    {
        name: 'message',
        baseName: 'message',
        type: 'string',
    },
    {
        name: 'context',
        baseName: 'context',
        type: 'object',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'number',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
