// @mirohq/miro-api/dist/model/organizationMember.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationMember = void 0;
/**
 * @internal
 * Organization member
 */
class OrganizationMember {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'organization-member';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return OrganizationMember.attributeTypeMap;
    }
}
exports.OrganizationMember = OrganizationMember;
/** @ignore */
OrganizationMember.discriminator = undefined;
/** @ignore */
OrganizationMember.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'active',
        baseName: 'active',
        type: 'boolean',
    },
    {
        name: 'email',
        baseName: 'email',
        type: 'string',
    },
    {
        name: 'lastActivityAt',
        baseName: 'lastActivityAt',
        type: 'Date',
    },
    {
        name: 'license',
        baseName: 'license',
        type: 'OrganizationMember.LicenseEnum',
    },
    {
        name: 'licenseAssignedAt',
        baseName: 'licenseAssignedAt',
        type: 'Date',
    },
    {
        name: 'role',
        baseName: 'role',
        type: 'OrganizationMember.RoleEnum',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
(function (OrganizationMember) {
    OrganizationMember.LicenseEnum = {
        Full: 'full',
        Occasional: 'occasional',
        Free: 'free',
        FreeRestricted: 'free_restricted',
        FullTrial: 'full_trial',
        Unknown: 'unknown',
    };
    OrganizationMember.RoleEnum = {
        OrganizationInternalAdmin: 'organization_internal_admin',
        OrganizationInternalUser: 'organization_internal_user',
        OrganizationExternalUser: 'organization_external_user',
        OrganizationTeamGuestUser: 'organization_team_guest_user',
        Unknown: 'unknown',
    };
})(OrganizationMember = exports.OrganizationMember || (exports.OrganizationMember = {}));
