// @mirohq/miro-api/dist/model/organization.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organization = void 0;
class Organization {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'organization';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return Organization.attributeTypeMap;
    }
}
exports.Organization = Organization;
/** @ignore */
Organization.discriminator = undefined;
/** @ignore */
Organization.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'fullLicensesPurchased',
        baseName: 'fullLicensesPurchased',
        type: 'number',
    },
    {
        name: 'name',
        baseName: 'name',
        type: 'string',
    },
    {
        name: 'plan',
        baseName: 'plan',
        type: 'Organization.PlanEnum',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
(function (Organization) {
    Organization.PlanEnum = {
        Company: 'company',
        Consultant: 'consultant',
        ConsultantSlf: 'consultant_slf',
        Business: 'business',
        PaidTeamOrg: 'paid_team_org',
        IntegrationOrg: 'integration_org',
        Professional2022: 'professional_2022',
        EduTeamOrg: 'edu_team_org',
        FreeTeamOrg: 'free_team_org',
        DevTeamOrg: 'dev_team_org',
        Unknown: 'unknown',
    };
})(Organization = exports.Organization || (exports.Organization = {}));
