// @mirohq/miro-api/dist/model/pageLinksPlatformContainers.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PageLinksPlatformContainers = void 0;
/**
 * @internal
 * Contains pagination links for the collection.
 */
class PageLinksPlatformContainers {
    /** @ignore */
    static getAttributeTypeMap() {
        return PageLinksPlatformContainers.attributeTypeMap;
    }
}
exports.PageLinksPlatformContainers = PageLinksPlatformContainers;
/** @ignore */
PageLinksPlatformContainers.discriminator = undefined;
/** @ignore */
PageLinksPlatformContainers.attributeTypeMap = [
    {
        name: 'first',
        baseName: 'first',
        type: 'string',
    },
    {
        name: 'last',
        baseName: 'last',
        type: 'string',
    },
    {
        name: 'next',
        baseName: 'next',
        type: 'string',
    },
    {
        name: 'prev',
        baseName: 'prev',
        type: 'string',
    },
    {
        name: 'self',
        baseName: 'self',
        type: 'string',
    },
];
