// @mirohq/miro-api/dist/model/picture.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Picture = void 0;
class Picture {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'picture';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return Picture.attributeTypeMap;
    }
}
exports.Picture = Picture;
/** @ignore */
Picture.discriminator = undefined;
/** @ignore */
Picture.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'imageURL',
        baseName: 'imageURL',
        type: 'string',
    },
    {
        name: 'originalUrl',
        baseName: 'originalUrl',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
