// @mirohq/miro-api/dist/model/position.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Position = void 0;
/**
 * @internal
 * Contains location information about the item, such as its x coordinate, y coordinate, and the origin of the x and y coordinates.
 */
class Position {
    constructor() {
        /**
         * Area of the item that is referenced by its x and y coordinates. For example, an item with a center origin will have its x and y coordinates point to its center. The center point of the board has x: 0 and y: 0 coordinates. Currently, only one option is supported.
         */
        this['origin'] = Position.OriginEnum.Center;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return Position.attributeTypeMap;
    }
}
exports.Position = Position;
/** @ignore */
Position.discriminator = undefined;
/** @ignore */
Position.attributeTypeMap = [
    {
        name: 'origin',
        baseName: 'origin',
        type: 'Position.OriginEnum',
    },
    {
        name: 'relativeTo',
        baseName: 'relativeTo',
        type: 'Position.RelativeToEnum',
    },
    {
        name: 'x',
        baseName: 'x',
        type: 'number',
    },
    {
        name: 'y',
        baseName: 'y',
        type: 'number',
    },
];
(function (Position) {
    Position.OriginEnum = {
        Center: 'center',
    };
    Position.RelativeToEnum = {
        CanvasCenter: 'canvas_center',
        ParentTopLeft: 'parent_top_left',
    };
})(Position = exports.Position || (exports.Position = {}));
