// @mirohq/miro-api/dist/model/projectMember.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectMember = void 0;
class ProjectMember {
    constructor() {
        /**
         * Type of the object
         */
        this['type'] = 'project_member';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return ProjectMember.attributeTypeMap;
    }
}
exports.ProjectMember = ProjectMember;
/** @ignore */
ProjectMember.discriminator = undefined;
/** @ignore */
ProjectMember.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'email',
        baseName: 'email',
        type: 'string',
    },
    {
        name: 'role',
        baseName: 'role',
        type: 'ProjectRole',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
