// @mirohq/miro-api/dist/model/projectPage.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectPage = void 0;
class ProjectPage {
    constructor() {
        /**
         * The maximum number of results to return per call. If the number of project in the response is greater than the limit specified, the response returns the cursor parameter with a value.
         */
        this['limit'] = 100;
        /**
         * Number of results returned in the response considering the cursor and the limit values sent in the request. For example, if there are 20 results, the request does not have a cursor value, and the limit set to 10, the size of the results will be 10. In this example, the response will also return a cursor value that can be used to retrieve the next set of 10 remaining results in the collection.
         */
        this['size'] = 100;
        /**
         * Type of the object returned.
         */
        this['type'] = 'cursor-list';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return ProjectPage.attributeTypeMap;
    }
}
exports.ProjectPage = ProjectPage;
/** @ignore */
ProjectPage.discriminator = undefined;
/** @ignore */
ProjectPage.attributeTypeMap = [
    {
        name: 'limit',
        baseName: 'limit',
        type: 'number',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'data',
        baseName: 'data',
        type: 'Array<Project>',
    },
    {
        name: 'cursor',
        baseName: 'cursor',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
