// @mirohq/miro-api/dist/model/relativeOffset.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RelativeOffset = void 0;
/**
 * @internal
 * The relative position of the point on an item where the connector is attached. Position with x=0% and y=0% correspond to the top-left corner of the item, x=100% and y=100% correspond to the right-bottom corner.
 */
class RelativeOffset {
    /** @ignore */
    static getAttributeTypeMap() {
        return RelativeOffset.attributeTypeMap;
    }
}
exports.RelativeOffset = RelativeOffset;
/** @ignore */
RelativeOffset.discriminator = undefined;
/** @ignore */
RelativeOffset.attributeTypeMap = [
    {
        name: 'x',
        baseName: 'x',
        type: 'string',
    },
    {
        name: 'y',
        baseName: 'y',
        type: 'string',
    },
];
