// @mirohq/miro-api/dist/model/shapeCreateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShapeCreateRequest = void 0;
class ShapeCreateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return ShapeCreateRequest.attributeTypeMap;
    }
}
exports.ShapeCreateRequest = ShapeCreateRequest;
/** @ignore */
ShapeCreateRequest.discriminator = undefined;
/** @ignore */
ShapeCreateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'ShapeDataForCreate',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'ShapeStyleForCreate',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'Geometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
