// @mirohq/miro-api/dist/model/shapeStyleForCreate.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShapeStyleForCreate = void 0;
/**
 * @internal
 * Contains information about the shape style, such as the border color or opacity. <br> All properties in style object are supported for shape types aren\'t listed below. <br> <table>   <tr>     <th align=\"left\">Shape type</th>     <th align=\"left\">Unsupported properties</th>   </tr>   <tr>     <td>flow_chart_or</td>     <td>fontSize, fontFamily, color, textAlign, textAlignVertical</td>   </tr>   <tr>     <td>flow_chart_summing_junction</td>     <td>fontSize, fontFamily, color, textAlign, textAlignVertical</td>   </tr>   <tr>     <td>flow_chart_note_curly_left</td>     <td>fillColor, fillOpacity</td>   </tr>   <tr>     <td>flow_chart_note_curly_right</td>     <td>fillColor, fillOpacity</td>   </tr>   <tr>     <td>flow_chart_note_square</td>     <td>fillColor, fillOpacity</td>   </tr> </table>
 */
class ShapeStyleForCreate {
    /** @ignore */
    static getAttributeTypeMap() {
        return ShapeStyleForCreate.attributeTypeMap;
    }
}
exports.ShapeStyleForCreate = ShapeStyleForCreate;
/** @ignore */
ShapeStyleForCreate.discriminator = undefined;
/** @ignore */
ShapeStyleForCreate.attributeTypeMap = [
    {
        name: 'borderColor',
        baseName: 'borderColor',
        type: 'string',
    },
    {
        name: 'borderOpacity',
        baseName: 'borderOpacity',
        type: 'string',
    },
    {
        name: 'borderStyle',
        baseName: 'borderStyle',
        type: 'ShapeStyleForCreate.BorderStyleEnum',
    },
    {
        name: 'borderWidth',
        baseName: 'borderWidth',
        type: 'string',
    },
    {
        name: 'color',
        baseName: 'color',
        type: 'string',
    },
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'string',
    },
    {
        name: 'fillOpacity',
        baseName: 'fillOpacity',
        type: 'string',
    },
    {
        name: 'fontFamily',
        baseName: 'fontFamily',
        type: 'ShapeStyleForCreate.FontFamilyEnum',
    },
    {
        name: 'fontSize',
        baseName: 'fontSize',
        type: 'string',
    },
    {
        name: 'textAlign',
        baseName: 'textAlign',
        type: 'ShapeStyleForCreate.TextAlignEnum',
    },
    {
        name: 'textAlignVertical',
        baseName: 'textAlignVertical',
        type: 'ShapeStyleForCreate.TextAlignVerticalEnum',
    },
];
(function (ShapeStyleForCreate) {
    ShapeStyleForCreate.BorderStyleEnum = {
        Normal: 'normal',
        Dotted: 'dotted',
        Dashed: 'dashed',
    };
    ShapeStyleForCreate.FontFamilyEnum = {
        Arial: 'arial',
        AbrilFatface: 'abril_fatface',
        Bangers: 'bangers',
        EbGaramond: 'eb_garamond',
        Georgia: 'georgia',
        Graduate: 'graduate',
        GravitasOne: 'gravitas_one',
        FredokaOne: 'fredoka_one',
        NixieOne: 'nixie_one',
        OpenSans: 'open_sans',
        PermanentMarker: 'permanent_marker',
        PtSans: 'pt_sans',
        PtSansNarrow: 'pt_sans_narrow',
        PtSerif: 'pt_serif',
        RammettoOne: 'rammetto_one',
        Roboto: 'roboto',
        RobotoCondensed: 'roboto_condensed',
        RobotoSlab: 'roboto_slab',
        Caveat: 'caveat',
        TimesNewRoman: 'times_new_roman',
        TitanOne: 'titan_one',
        LemonTuesday: 'lemon_tuesday',
        RobotoMono: 'roboto_mono',
        NotoSans: 'noto_sans',
        PlexSans: 'plex_sans',
        PlexSerif: 'plex_serif',
        PlexMono: 'plex_mono',
        Spoof: 'spoof',
        TiemposText: 'tiempos_text',
        Formular: 'formular',
    };
    ShapeStyleForCreate.TextAlignEnum = {
        Left: 'left',
        Right: 'right',
        Center: 'center',
        Unknown: 'unknown',
    };
    ShapeStyleForCreate.TextAlignVerticalEnum = {
        Top: 'top',
        Middle: 'middle',
        Bottom: 'bottom',
        Unknown: 'unknown',
    };
})(ShapeStyleForCreate = exports.ShapeStyleForCreate || (exports.ShapeStyleForCreate = {}));
