// @mirohq/miro-api/dist/model/shapeUpdateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShapeUpdateRequest = void 0;
class ShapeUpdateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return ShapeUpdateRequest.attributeTypeMap;
    }
}
exports.ShapeUpdateRequest = ShapeUpdateRequest;
/** @ignore */
ShapeUpdateRequest.discriminator = undefined;
/** @ignore */
ShapeUpdateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'ShapeDataForUpdate',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'ShapeStyleForUpdate',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'Geometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
