// @mirohq/miro-api/dist/model/shapeStyle.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShapeStyle = void 0;
/**
 * @internal
 * Contains information about the shape style, such as the border color or opacity.
 */
class ShapeStyle {
    /** @ignore */
    static getAttributeTypeMap() {
        return ShapeStyle.attributeTypeMap;
    }
}
exports.ShapeStyle = ShapeStyle;
/** @ignore */
ShapeStyle.discriminator = undefined;
/** @ignore */
ShapeStyle.attributeTypeMap = [
    {
        name: 'borderColor',
        baseName: 'borderColor',
        type: 'string',
    },
    {
        name: 'borderOpacity',
        baseName: 'borderOpacity',
        type: 'string',
    },
    {
        name: 'borderStyle',
        baseName: 'borderStyle',
        type: 'ShapeStyle.BorderStyleEnum',
    },
    {
        name: 'borderWidth',
        baseName: 'borderWidth',
        type: 'string',
    },
    {
        name: 'color',
        baseName: 'color',
        type: 'string',
    },
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'string',
    },
    {
        name: 'fillOpacity',
        baseName: 'fillOpacity',
        type: 'string',
    },
    {
        name: 'fontFamily',
        baseName: 'fontFamily',
        type: 'ShapeStyle.FontFamilyEnum',
    },
    {
        name: 'fontSize',
        baseName: 'fontSize',
        type: 'string',
    },
    {
        name: 'textAlign',
        baseName: 'textAlign',
        type: 'ShapeStyle.TextAlignEnum',
    },
    {
        name: 'textAlignVertical',
        baseName: 'textAlignVertical',
        type: 'ShapeStyle.TextAlignVerticalEnum',
    },
];
(function (ShapeStyle) {
    ShapeStyle.BorderStyleEnum = {
        Normal: 'normal',
        Dotted: 'dotted',
        Dashed: 'dashed',
    };
    ShapeStyle.FontFamilyEnum = {
        Arial: 'arial',
        AbrilFatface: 'abril_fatface',
        Bangers: 'bangers',
        EbGaramond: 'eb_garamond',
        Georgia: 'georgia',
        Graduate: 'graduate',
        GravitasOne: 'gravitas_one',
        FredokaOne: 'fredoka_one',
        NixieOne: 'nixie_one',
        OpenSans: 'open_sans',
        PermanentMarker: 'permanent_marker',
        PtSans: 'pt_sans',
        PtSansNarrow: 'pt_sans_narrow',
        PtSerif: 'pt_serif',
        RammettoOne: 'rammetto_one',
        Roboto: 'roboto',
        RobotoCondensed: 'roboto_condensed',
        RobotoSlab: 'roboto_slab',
        Caveat: 'caveat',
        TimesNewRoman: 'times_new_roman',
        TitanOne: 'titan_one',
        LemonTuesday: 'lemon_tuesday',
        RobotoMono: 'roboto_mono',
        NotoSans: 'noto_sans',
        PlexSans: 'plex_sans',
        PlexSerif: 'plex_serif',
        PlexMono: 'plex_mono',
        Spoof: 'spoof',
        TiemposText: 'tiempos_text',
        Formular: 'formular',
    };
    ShapeStyle.TextAlignEnum = {
        Left: 'left',
        Right: 'right',
        Center: 'center',
    };
    ShapeStyle.TextAlignVerticalEnum = {
        Top: 'top',
        Middle: 'middle',
        Bottom: 'bottom',
    };
})(ShapeStyle = exports.ShapeStyle || (exports.ShapeStyle = {}));
