// @mirohq/miro-api/dist/model/stickyNoteStyle.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StickyNoteStyle = void 0;
/**
 * @internal
 * Contains information about the style of a sticky note item, such as the fill color or text alignment.
 */
class StickyNoteStyle {
    /** @ignore */
    static getAttributeTypeMap() {
        return StickyNoteStyle.attributeTypeMap;
    }
}
exports.StickyNoteStyle = StickyNoteStyle;
/** @ignore */
StickyNoteStyle.discriminator = undefined;
/** @ignore */
StickyNoteStyle.attributeTypeMap = [
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'StickyNoteStyle.FillColorEnum',
    },
    {
        name: 'textAlign',
        baseName: 'textAlign',
        type: 'StickyNoteStyle.TextAlignEnum',
    },
    {
        name: 'textAlignVertical',
        baseName: 'textAlignVertical',
        type: 'StickyNoteStyle.TextAlignVerticalEnum',
    },
];
(function (StickyNoteStyle) {
    StickyNoteStyle.FillColorEnum = {
        Gray: 'gray',
        LightYellow: 'light_yellow',
        Yellow: 'yellow',
        Orange: 'orange',
        LightGreen: 'light_green',
        Green: 'green',
        DarkGreen: 'dark_green',
        Cyan: 'cyan',
        LightPink: 'light_pink',
        Pink: 'pink',
        Violet: 'violet',
        Red: 'red',
        LightBlue: 'light_blue',
        Blue: 'blue',
        DarkBlue: 'dark_blue',
        Black: 'black',
    };
    StickyNoteStyle.TextAlignEnum = {
        Left: 'left',
        Right: 'right',
        Center: 'center',
    };
    StickyNoteStyle.TextAlignVerticalEnum = {
        Top: 'top',
        Middle: 'middle',
        Bottom: 'bottom',
    };
})(StickyNoteStyle = exports.StickyNoteStyle || (exports.StickyNoteStyle = {}));
