// @mirohq/miro-api/dist/model/stickyNoteDataPlatformBulkCreateOperationExperimentalRelease.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StickyNoteDataPlatformBulkCreateOperationExperimentalRelease = void 0;
/**
 * @internal
 * Contains sticky note item data, such as the content or shape of the sticky note.
 */
class StickyNoteDataPlatformBulkCreateOperationExperimentalRelease {
    constructor() {
        /**
         * Defines the geometric shape of the sticky note and aspect ratio for its dimensions.
         */
        this['shape'] = StickyNoteDataPlatformBulkCreateOperationExperimentalRelease.ShapeEnum.Square;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return StickyNoteDataPlatformBulkCreateOperationExperimentalRelease.attributeTypeMap;
    }
}
exports.StickyNoteDataPlatformBulkCreateOperationExperimentalRelease = StickyNoteDataPlatformBulkCreateOperationExperimentalRelease;
/** @ignore */
StickyNoteDataPlatformBulkCreateOperationExperimentalRelease.discriminator = undefined;
/** @ignore */
StickyNoteDataPlatformBulkCreateOperationExperimentalRelease.attributeTypeMap = [
    {
        name: 'content',
        baseName: 'content',
        type: 'string',
    },
    {
        name: 'shape',
        baseName: 'shape',
        type: 'StickyNoteDataPlatformBulkCreateOperationExperimentalRelease.ShapeEnum',
    },
];
(function (StickyNoteDataPlatformBulkCreateOperationExperimentalRelease) {
    StickyNoteDataPlatformBulkCreateOperationExperimentalRelease.ShapeEnum = {
        Square: 'square',
        Rectangle: 'rectangle',
    };
})(StickyNoteDataPlatformBulkCreateOperationExperimentalRelease = exports.StickyNoteDataPlatformBulkCreateOperationExperimentalRelease || (exports.StickyNoteDataPlatformBulkCreateOperationExperimentalRelease = {}));
