// @mirohq/miro-api/dist/model/stickyNoteUpdateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StickyNoteUpdateRequest = void 0;
class StickyNoteUpdateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return StickyNoteUpdateRequest.attributeTypeMap;
    }
}
exports.StickyNoteUpdateRequest = StickyNoteUpdateRequest;
/** @ignore */
StickyNoteUpdateRequest.discriminator = undefined;
/** @ignore */
StickyNoteUpdateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'StickyNoteData',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'StickyNoteStyle',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'FixedRatioNoRotationGeometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
