// @mirohq/miro-api/dist/model/tag.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tag = void 0;
class Tag {
    /** @ignore */
    static getAttributeTypeMap() {
        return Tag.attributeTypeMap;
    }
}
exports.Tag = Tag;
/** @ignore */
Tag.discriminator = undefined;
/** @ignore */
Tag.attributeTypeMap = [
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'Tag.FillColorEnum',
    },
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
(function (Tag) {
    Tag.FillColorEnum = {
        Red: 'red',
        LightGreen: 'light_green',
        Cyan: 'cyan',
        Yellow: 'yellow',
        Magenta: 'magenta',
        Green: 'green',
        Blue: 'blue',
        Gray: 'gray',
        Violet: 'violet',
        DarkGreen: 'dark_green',
        DarkBlue: 'dark_blue',
        Black: 'black',
    };
})(Tag = exports.Tag || (exports.Tag = {}));
