// @mirohq/miro-api/dist/model/tagCreateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagCreateRequest = void 0;
class TagCreateRequest {
    constructor() {
        /**
         * Fill color for the tag.
         */
        this['fillColor'] = TagCreateRequest.FillColorEnum.Red;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return TagCreateRequest.attributeTypeMap;
    }
}
exports.TagCreateRequest = TagCreateRequest;
/** @ignore */
TagCreateRequest.discriminator = undefined;
/** @ignore */
TagCreateRequest.attributeTypeMap = [
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'TagCreateRequest.FillColorEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
];
(function (TagCreateRequest) {
    TagCreateRequest.FillColorEnum = {
        Red: 'red',
        LightGreen: 'light_green',
        Cyan: 'cyan',
        Yellow: 'yellow',
        Magenta: 'magenta',
        Green: 'green',
        Blue: 'blue',
        Gray: 'gray',
        Violet: 'violet',
        DarkGreen: 'dark_green',
        DarkBlue: 'dark_blue',
        Black: 'black',
    };
})(TagCreateRequest = exports.TagCreateRequest || (exports.TagCreateRequest = {}));
