// @mirohq/miro-api/dist/model/tagsPagedResponse.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagsPagedResponse = void 0;
class TagsPagedResponse {
    /** @ignore */
    static getAttributeTypeMap() {
        return TagsPagedResponse.attributeTypeMap;
    }
}
exports.TagsPagedResponse = TagsPagedResponse;
/** @ignore */
TagsPagedResponse.discriminator = undefined;
/** @ignore */
TagsPagedResponse.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'Array<Tag>',
    },
    {
        name: 'total',
        baseName: 'total',
        type: 'number',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'offset',
        baseName: 'offset',
        type: 'number',
    },
    {
        name: 'limit',
        baseName: 'limit',
        type: 'number',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'PageLinks',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
