// @mirohq/miro-api/dist/model/tagUpdateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagUpdateRequest = void 0;
class TagUpdateRequest {
    constructor() {
        /**
         * Fill color for the tag.
         */
        this['fillColor'] = TagUpdateRequest.FillColorEnum.Red;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return TagUpdateRequest.attributeTypeMap;
    }
}
exports.TagUpdateRequest = TagUpdateRequest;
/** @ignore */
TagUpdateRequest.discriminator = undefined;
/** @ignore */
TagUpdateRequest.attributeTypeMap = [
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'TagUpdateRequest.FillColorEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
];
(function (TagUpdateRequest) {
    TagUpdateRequest.FillColorEnum = {
        Red: 'red',
        LightGreen: 'light_green',
        Cyan: 'cyan',
        Yellow: 'yellow',
        Magenta: 'magenta',
        Green: 'green',
        Blue: 'blue',
        Gray: 'gray',
        Violet: 'violet',
        DarkGreen: 'dark_green',
        DarkBlue: 'dark_blue',
        Black: 'black',
    };
})(TagUpdateRequest = exports.TagUpdateRequest || (exports.TagUpdateRequest = {}));
