// @mirohq/miro-api/dist/model/tagWithLinks.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagWithLinks = void 0;
class TagWithLinks {
    /** @ignore */
    static getAttributeTypeMap() {
        return TagWithLinks.attributeTypeMap;
    }
}
exports.TagWithLinks = TagWithLinks;
/** @ignore */
TagWithLinks.discriminator = undefined;
/** @ignore */
TagWithLinks.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'TagWithLinks.FillColorEnum',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'SelfLink',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
(function (TagWithLinks) {
    TagWithLinks.FillColorEnum = {
        Red: 'red',
        LightGreen: 'light_green',
        Cyan: 'cyan',
        Yellow: 'yellow',
        Magenta: 'magenta',
        Green: 'green',
        Blue: 'blue',
        Gray: 'gray',
        Violet: 'violet',
        DarkGreen: 'dark_green',
        DarkBlue: 'dark_blue',
        Black: 'black',
    };
})(TagWithLinks = exports.TagWithLinks || (exports.TagWithLinks = {}));
