// @mirohq/miro-api/dist/model/teamCopyAccessLevelSettings.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamCopyAccessLevelSettings = void 0;
/**
 * @internal
 * Team copy access settings
 */
class TeamCopyAccessLevelSettings {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamCopyAccessLevelSettings.attributeTypeMap;
    }
}
exports.TeamCopyAccessLevelSettings = TeamCopyAccessLevelSettings;
/** @ignore */
TeamCopyAccessLevelSettings.discriminator = undefined;
/** @ignore */
TeamCopyAccessLevelSettings.attributeTypeMap = [
    {
        name: 'copyAccessLevel',
        baseName: 'copyAccessLevel',
        type: 'TeamCopyAccessLevelSettings.CopyAccessLevelEnum',
    },
    {
        name: 'copyAccessLevelLimitation',
        baseName: 'copyAccessLevelLimitation',
        type: 'TeamCopyAccessLevelSettings.CopyAccessLevelLimitationEnum',
    },
];
(function (TeamCopyAccessLevelSettings) {
    TeamCopyAccessLevelSettings.CopyAccessLevelEnum = {
        Anyone: 'anyone',
        TeamMembers: 'team_members',
        TeamEditors: 'team_editors',
        BoardOwner: 'board_owner            -',
    };
    TeamCopyAccessLevelSettings.CopyAccessLevelLimitationEnum = {
        Anyone: 'anyone',
        TeamMembers: 'team_members',
    };
})(TeamCopyAccessLevelSettings = exports.TeamCopyAccessLevelSettings || (exports.TeamCopyAccessLevelSettings = {}));
