// @mirohq/miro-api/dist/model/teamMemberInvite.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamMemberInvite = void 0;
class TeamMemberInvite {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamMemberInvite.attributeTypeMap;
    }
}
exports.TeamMemberInvite = TeamMemberInvite;
/** @ignore */
TeamMemberInvite.discriminator = undefined;
/** @ignore */
TeamMemberInvite.attributeTypeMap = [
    {
        name: 'email',
        baseName: 'email',
        type: 'string',
    },
    {
        name: 'role',
        baseName: 'role',
        type: 'TeamMemberInvite.RoleEnum',
    },
];
(function (TeamMemberInvite) {
    TeamMemberInvite.RoleEnum = {
        Member: 'member',
        Admin: 'admin',
        TeamGuest: 'team_guest',
    };
})(TeamMemberInvite = exports.TeamMemberInvite || (exports.TeamMemberInvite = {}));
