// @mirohq/miro-api/dist/model/teamMemberChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamMemberChanges = void 0;
class TeamMemberChanges {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamMemberChanges.attributeTypeMap;
    }
}
exports.TeamMemberChanges = TeamMemberChanges;
/** @ignore */
TeamMemberChanges.discriminator = undefined;
/** @ignore */
TeamMemberChanges.attributeTypeMap = [
    {
        name: 'role',
        baseName: 'role',
        type: 'TeamMemberChanges.RoleEnum',
    },
];
(function (TeamMemberChanges) {
    TeamMemberChanges.RoleEnum = {
        Member: 'member',
        Admin: 'admin',
        TeamGuest: 'team_guest',
    };
})(TeamMemberChanges = exports.TeamMemberChanges || (exports.TeamMemberChanges = {}));
