// @mirohq/miro-api/dist/model/teamAccountDiscoverySettings.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamAccountDiscoverySettings = void 0;
/**
 * @internal
 * Team account discovery settings
 */
class TeamAccountDiscoverySettings {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamAccountDiscoverySettings.attributeTypeMap;
    }
}
exports.TeamAccountDiscoverySettings = TeamAccountDiscoverySettings;
/** @ignore */
TeamAccountDiscoverySettings.discriminator = undefined;
/** @ignore */
TeamAccountDiscoverySettings.attributeTypeMap = [
    {
        name: 'accountDiscovery',
        baseName: 'accountDiscovery',
        type: 'TeamAccountDiscoverySettings.AccountDiscoveryEnum',
    },
];
(function (TeamAccountDiscoverySettings) {
    TeamAccountDiscoverySettings.AccountDiscoveryEnum = {
        Hidden: 'hidden',
        Request: 'request',
        Join: 'join',
    };
})(TeamAccountDiscoverySettings = exports.TeamAccountDiscoverySettings || (exports.TeamAccountDiscoverySettings = {}));
