// @mirohq/miro-api/dist/model/teamAccountDiscoverySettingsChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamAccountDiscoverySettingsChanges = void 0;
/**
 * @internal
 * Team account discovery settings
 */
class TeamAccountDiscoverySettingsChanges {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamAccountDiscoverySettingsChanges.attributeTypeMap;
    }
}
exports.TeamAccountDiscoverySettingsChanges = TeamAccountDiscoverySettingsChanges;
/** @ignore */
TeamAccountDiscoverySettingsChanges.discriminator = undefined;
/** @ignore */
TeamAccountDiscoverySettingsChanges.attributeTypeMap = [
    {
        name: 'accountDiscovery',
        baseName: 'accountDiscovery',
        type: 'TeamAccountDiscoverySettingsChanges.AccountDiscoveryEnum',
    },
];
(function (TeamAccountDiscoverySettingsChanges) {
    TeamAccountDiscoverySettingsChanges.AccountDiscoveryEnum = {
        Hidden: 'hidden',
        Request: 'request',
        Join: 'join',
    };
})(TeamAccountDiscoverySettingsChanges = exports.TeamAccountDiscoverySettingsChanges || (exports.TeamAccountDiscoverySettingsChanges = {}));
