// @mirohq/miro-api/dist/model/teamCollaborationSettingsChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamCollaborationSettingsChanges = void 0;
/**
 * @internal
 * Team collaboration settings
 */
class TeamCollaborationSettingsChanges {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamCollaborationSettingsChanges.attributeTypeMap;
    }
}
exports.TeamCollaborationSettingsChanges = TeamCollaborationSettingsChanges;
/** @ignore */
TeamCollaborationSettingsChanges.discriminator = undefined;
/** @ignore */
TeamCollaborationSettingsChanges.attributeTypeMap = [
    {
        name: 'coOwnerRole',
        baseName: 'coOwnerRole',
        type: 'TeamCollaborationSettingsChanges.CoOwnerRoleEnum',
    },
];
(function (TeamCollaborationSettingsChanges) {
    TeamCollaborationSettingsChanges.CoOwnerRoleEnum = {
        Enabled: 'enabled',
        Disabled: 'disabled',
    };
})(TeamCollaborationSettingsChanges = exports.TeamCollaborationSettingsChanges || (exports.TeamCollaborationSettingsChanges = {}));
