// @mirohq/miro-api/dist/model/teamInvitationSettings.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamInvitationSettings = void 0;
/**
 * @internal
 * Team invitation settings
 */
class TeamInvitationSettings {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamInvitationSettings.attributeTypeMap;
    }
}
exports.TeamInvitationSettings = TeamInvitationSettings;
/** @ignore */
TeamInvitationSettings.discriminator = undefined;
/** @ignore */
TeamInvitationSettings.attributeTypeMap = [
    {
        name: 'inviteExternalUsers',
        baseName: 'inviteExternalUsers',
        type: 'TeamInvitationSettings.InviteExternalUsersEnum',
    },
    {
        name: 'whoCanInvite',
        baseName: 'whoCanInvite',
        type: 'TeamInvitationSettings.WhoCanInviteEnum',
    },
];
(function (TeamInvitationSettings) {
    TeamInvitationSettings.InviteExternalUsersEnum = {
        Allowed: 'allowed',
        NotAllowed: 'not_allowed',
    };
    TeamInvitationSettings.WhoCanInviteEnum = {
        OnlyOrgAdmins: 'only_org_admins',
        Admins: 'admins',
        AllMembers: 'all_members',
    };
})(TeamInvitationSettings = exports.TeamInvitationSettings || (exports.TeamInvitationSettings = {}));
