// @mirohq/miro-api/dist/model/teamSharingPolicySettingsChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamSharingPolicySettingsChanges = void 0;
/**
 * @internal
 * Team sharing policy settings
 */
class TeamSharingPolicySettingsChanges {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamSharingPolicySettingsChanges.attributeTypeMap;
    }
}
exports.TeamSharingPolicySettingsChanges = TeamSharingPolicySettingsChanges;
/** @ignore */
TeamSharingPolicySettingsChanges.discriminator = undefined;
/** @ignore */
TeamSharingPolicySettingsChanges.attributeTypeMap = [
    {
        name: 'allowListedDomains',
        baseName: 'allowListedDomains',
        type: 'Array<string>',
    },
    {
        name: 'createAssetAccessLevel',
        baseName: 'createAssetAccessLevel',
        type: 'TeamSharingPolicySettingsChanges.CreateAssetAccessLevelEnum',
    },
    {
        name: 'defaultBoardAccess',
        baseName: 'defaultBoardAccess',
        type: 'TeamSharingPolicySettingsChanges.DefaultBoardAccessEnum',
    },
    {
        name: 'defaultOrganizationAccess',
        baseName: 'defaultOrganizationAccess',
        type: 'TeamSharingPolicySettingsChanges.DefaultOrganizationAccessEnum',
    },
    {
        name: 'defaultProjectAccess',
        baseName: 'defaultProjectAccess',
        type: 'TeamSharingPolicySettingsChanges.DefaultProjectAccessEnum',
    },
    {
        name: 'moveBoardToAccount',
        baseName: 'moveBoardToAccount',
        type: 'TeamSharingPolicySettingsChanges.MoveBoardToAccountEnum',
    },
    {
        name: 'restrictAllowedDomains',
        baseName: 'restrictAllowedDomains',
        type: 'TeamSharingPolicySettingsChanges.RestrictAllowedDomainsEnum',
    },
    {
        name: 'sharingOnAccount',
        baseName: 'sharingOnAccount',
        type: 'TeamSharingPolicySettingsChanges.SharingOnAccountEnum',
    },
    {
        name: 'sharingOnOrganization',
        baseName: 'sharingOnOrganization',
        type: 'TeamSharingPolicySettingsChanges.SharingOnOrganizationEnum',
    },
    {
        name: 'sharingViaPublicLink',
        baseName: 'sharingViaPublicLink',
        type: 'TeamSharingPolicySettingsChanges.SharingViaPublicLinkEnum',
    },
];
(function (TeamSharingPolicySettingsChanges) {
    TeamSharingPolicySettingsChanges.CreateAssetAccessLevelEnum = {
        CompanyAdmins: 'company_admins',
        Admins: 'admins',
        AllMembers: 'all_members',
    };
    TeamSharingPolicySettingsChanges.DefaultBoardAccessEnum = {
        Private: 'private',
        View: 'view',
        Comment: 'comment',
        Edit: 'edit',
    };
    TeamSharingPolicySettingsChanges.DefaultOrganizationAccessEnum = {
        Private: 'private',
        View: 'view',
        Comment: 'comment',
        Edit: 'edit',
    };
    TeamSharingPolicySettingsChanges.DefaultProjectAccessEnum = {
        Private: 'private',
        View: 'view',
    };
    TeamSharingPolicySettingsChanges.MoveBoardToAccountEnum = {
        Allowed: 'allowed',
        NotAllowed: 'not_allowed',
    };
    TeamSharingPolicySettingsChanges.RestrictAllowedDomainsEnum = {
        Enabled: 'enabled',
        EnabledWithExternalUserAccess: 'enabled_with_external_user_access',
        Disabled: 'disabled',
    };
    TeamSharingPolicySettingsChanges.SharingOnAccountEnum = {
        Allowed: 'allowed',
        NotAllowed: 'not_allowed',
    };
    TeamSharingPolicySettingsChanges.SharingOnOrganizationEnum = {
        Allowed: 'allowed',
        AllowedWithEditing: 'allowed_with_editing',
        NotAllowed: 'not_allowed',
    };
    TeamSharingPolicySettingsChanges.SharingViaPublicLinkEnum = {
        Allowed: 'allowed',
        AllowedWithEditing: 'allowed_with_editing',
        NotAllowed: 'not_allowed',
    };
})(TeamSharingPolicySettingsChanges = exports.TeamSharingPolicySettingsChanges || (exports.TeamSharingPolicySettingsChanges = {}));
