// @mirohq/miro-api/dist/model/teamInvitationSettingsChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamInvitationSettingsChanges = void 0;
/**
 * @internal
 * Team invitation settings
 */
class TeamInvitationSettingsChanges {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamInvitationSettingsChanges.attributeTypeMap;
    }
}
exports.TeamInvitationSettingsChanges = TeamInvitationSettingsChanges;
/** @ignore */
TeamInvitationSettingsChanges.discriminator = undefined;
/** @ignore */
TeamInvitationSettingsChanges.attributeTypeMap = [
    {
        name: 'inviteExternalUsers',
        baseName: 'inviteExternalUsers',
        type: 'TeamInvitationSettingsChanges.InviteExternalUsersEnum',
    },
    {
        name: 'whoCanInvite',
        baseName: 'whoCanInvite',
        type: 'TeamInvitationSettingsChanges.WhoCanInviteEnum',
    },
];
(function (TeamInvitationSettingsChanges) {
    TeamInvitationSettingsChanges.InviteExternalUsersEnum = {
        Allowed: 'allowed',
        NotAllowed: 'not_allowed',
    };
    TeamInvitationSettingsChanges.WhoCanInviteEnum = {
        OnlyOrgAdmins: 'only_org_admins',
        Admins: 'admins',
        AllMembers: 'all_members',
    };
})(TeamInvitationSettingsChanges = exports.TeamInvitationSettingsChanges || (exports.TeamInvitationSettingsChanges = {}));
