// @mirohq/miro-api/dist/model/teamSettings.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamSettings = void 0;
class TeamSettings {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'team-settings';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamSettings.attributeTypeMap;
    }
}
exports.TeamSettings = TeamSettings;
/** @ignore */
TeamSettings.discriminator = undefined;
/** @ignore */
TeamSettings.attributeTypeMap = [
    {
        name: 'organizationId',
        baseName: 'organizationId',
        type: 'string',
    },
    {
        name: 'teamAccountDiscoverySettings',
        baseName: 'teamAccountDiscoverySettings',
        type: 'TeamAccountDiscoverySettings',
    },
    {
        name: 'teamCollaborationSettings',
        baseName: 'teamCollaborationSettings',
        type: 'TeamCollaborationSettings',
    },
    {
        name: 'teamCopyAccessLevelSettings',
        baseName: 'teamCopyAccessLevelSettings',
        type: 'TeamCopyAccessLevelSettings',
    },
    {
        name: 'teamId',
        baseName: 'teamId',
        type: 'string',
    },
    {
        name: 'teamInvitationSettings',
        baseName: 'teamInvitationSettings',
        type: 'TeamInvitationSettings',
    },
    {
        name: 'teamSharingPolicySettings',
        baseName: 'teamSharingPolicySettings',
        type: 'TeamSharingPolicySettings',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
