// @mirohq/miro-api/dist/model/team.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Team = void 0;
class Team {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'team';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return Team.attributeTypeMap;
    }
}
exports.Team = Team;
/** @ignore */
Team.discriminator = undefined;
/** @ignore */
Team.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'name',
        baseName: 'name',
        type: 'string',
    },
    {
        name: 'picture',
        baseName: 'picture',
        type: 'Picture',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
