// @mirohq/miro-api/dist/model/teamMember.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamMember = void 0;
class TeamMember {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'team-member';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamMember.attributeTypeMap;
    }
}
exports.TeamMember = TeamMember;
/** @ignore */
TeamMember.discriminator = undefined;
/** @ignore */
TeamMember.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'role',
        baseName: 'role',
        type: 'TeamMember.RoleEnum',
    },
    {
        name: 'createdAt',
        baseName: 'createdAt',
        type: 'Date',
    },
    {
        name: 'createdBy',
        baseName: 'createdBy',
        type: 'string',
    },
    {
        name: 'modifiedAt',
        baseName: 'modifiedAt',
        type: 'Date',
    },
    {
        name: 'modifiedBy',
        baseName: 'modifiedBy',
        type: 'string',
    },
    {
        name: 'teamId',
        baseName: 'teamId',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
(function (TeamMember) {
    TeamMember.RoleEnum = {
        NonTeam: 'non_team',
        Member: 'member',
        Admin: 'admin',
        TeamGuest: 'team_guest',
    };
})(TeamMember = exports.TeamMember || (exports.TeamMember = {}));
