// @mirohq/miro-api/dist/model/teamCollaborationSettings.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamCollaborationSettings = void 0;
/**
 * @internal
 * Team collaboration settings
 */
class TeamCollaborationSettings {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamCollaborationSettings.attributeTypeMap;
    }
}
exports.TeamCollaborationSettings = TeamCollaborationSettings;
/** @ignore */
TeamCollaborationSettings.discriminator = undefined;
/** @ignore */
TeamCollaborationSettings.attributeTypeMap = [
    {
        name: 'coOwnerRole',
        baseName: 'coOwnerRole',
        type: 'TeamCollaborationSettings.CoOwnerRoleEnum',
    },
];
(function (TeamCollaborationSettings) {
    TeamCollaborationSettings.CoOwnerRoleEnum = {
        Enabled: 'enabled',
        Disabled: 'disabled',
    };
})(TeamCollaborationSettings = exports.TeamCollaborationSettings || (exports.TeamCollaborationSettings = {}));
