// @mirohq/miro-api/dist/model/teamCopyAccessLevelSettingsChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamCopyAccessLevelSettingsChanges = void 0;
/**
 * @internal
 * Team copy access settings
 */
class TeamCopyAccessLevelSettingsChanges {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamCopyAccessLevelSettingsChanges.attributeTypeMap;
    }
}
exports.TeamCopyAccessLevelSettingsChanges = TeamCopyAccessLevelSettingsChanges;
/** @ignore */
TeamCopyAccessLevelSettingsChanges.discriminator = undefined;
/** @ignore */
TeamCopyAccessLevelSettingsChanges.attributeTypeMap = [
    {
        name: 'copyAccessLevel',
        baseName: 'copyAccessLevel',
        type: 'TeamCopyAccessLevelSettingsChanges.CopyAccessLevelEnum',
    },
    {
        name: 'copyAccessLevelLimitation',
        baseName: 'copyAccessLevelLimitation',
        type: 'TeamCopyAccessLevelSettingsChanges.CopyAccessLevelLimitationEnum',
    },
];
(function (TeamCopyAccessLevelSettingsChanges) {
    TeamCopyAccessLevelSettingsChanges.CopyAccessLevelEnum = {
        Anyone: 'anyone',
        TeamMembers: 'team_members',
        TeamEditors: 'team_editors',
        BoardOwner: 'board_owner',
    };
    TeamCopyAccessLevelSettingsChanges.CopyAccessLevelLimitationEnum = {
        Anyone: 'anyone',
        TeamMembers: 'team_members',
    };
})(TeamCopyAccessLevelSettingsChanges = exports.TeamCopyAccessLevelSettingsChanges || (exports.TeamCopyAccessLevelSettingsChanges = {}));
