// @mirohq/miro-api/dist/model/teamSettingsChanges.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamSettingsChanges = void 0;
class TeamSettingsChanges {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamSettingsChanges.attributeTypeMap;
    }
}
exports.TeamSettingsChanges = TeamSettingsChanges;
/** @ignore */
TeamSettingsChanges.discriminator = undefined;
/** @ignore */
TeamSettingsChanges.attributeTypeMap = [
    {
        name: 'teamAccountDiscoverySettings',
        baseName: 'teamAccountDiscoverySettings',
        type: 'TeamAccountDiscoverySettingsChanges',
    },
    {
        name: 'teamCollaborationSettings',
        baseName: 'teamCollaborationSettings',
        type: 'TeamCollaborationSettingsChanges',
    },
    {
        name: 'teamCopyAccessLevelSettings',
        baseName: 'teamCopyAccessLevelSettings',
        type: 'TeamCopyAccessLevelSettingsChanges',
    },
    {
        name: 'teamInvitationSettings',
        baseName: 'teamInvitationSettings',
        type: 'TeamInvitationSettingsChanges',
    },
    {
        name: 'teamSharingPolicySettings',
        baseName: 'teamSharingPolicySettings',
        type: 'TeamSharingPolicySettingsChanges',
    },
];
