// @mirohq/miro-api/dist/model/textCreateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextCreateRequest = void 0;
class TextCreateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return TextCreateRequest.attributeTypeMap;
    }
}
exports.TextCreateRequest = TextCreateRequest;
/** @ignore */
TextCreateRequest.discriminator = undefined;
/** @ignore */
TextCreateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'TextData',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'TextStyle',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'WidthOnlyAdjustableGeometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
