// @mirohq/miro-api/dist/model/textStyle.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextStyle = void 0;
/**
 * @internal
 * Contains information about the style of a text item, such as the fill color or font family.
 */
class TextStyle {
    /** @ignore */
    static getAttributeTypeMap() {
        return TextStyle.attributeTypeMap;
    }
}
exports.TextStyle = TextStyle;
/** @ignore */
TextStyle.discriminator = undefined;
/** @ignore */
TextStyle.attributeTypeMap = [
    {
        name: 'color',
        baseName: 'color',
        type: 'string',
    },
    {
        name: 'fillColor',
        baseName: 'fillColor',
        type: 'string',
    },
    {
        name: 'fillOpacity',
        baseName: 'fillOpacity',
        type: 'string',
    },
    {
        name: 'fontFamily',
        baseName: 'fontFamily',
        type: 'TextStyle.FontFamilyEnum',
    },
    {
        name: 'fontSize',
        baseName: 'fontSize',
        type: 'string',
    },
    {
        name: 'textAlign',
        baseName: 'textAlign',
        type: 'TextStyle.TextAlignEnum',
    },
];
(function (TextStyle) {
    TextStyle.FontFamilyEnum = {
        Arial: 'arial',
        AbrilFatface: 'abril_fatface',
        Bangers: 'bangers',
        EbGaramond: 'eb_garamond',
        Georgia: 'georgia',
        Graduate: 'graduate',
        GravitasOne: 'gravitas_one',
        FredokaOne: 'fredoka_one',
        NixieOne: 'nixie_one',
        OpenSans: 'open_sans',
        PermanentMarker: 'permanent_marker',
        PtSans: 'pt_sans',
        PtSansNarrow: 'pt_sans_narrow',
        PtSerif: 'pt_serif',
        RammettoOne: 'rammetto_one',
        Roboto: 'roboto',
        RobotoCondensed: 'roboto_condensed',
        RobotoSlab: 'roboto_slab',
        Caveat: 'caveat',
        TimesNewRoman: 'times_new_roman',
        TitanOne: 'titan_one',
        LemonTuesday: 'lemon_tuesday',
        RobotoMono: 'roboto_mono',
        NotoSans: 'noto_sans',
        PlexSans: 'plex_sans',
        PlexSerif: 'plex_serif',
        PlexMono: 'plex_mono',
        Spoof: 'spoof',
        TiemposText: 'tiempos_text',
        Formular: 'formular',
    };
    TextStyle.TextAlignEnum = {
        Left: 'left',
        Right: 'right',
        Center: 'center',
    };
})(TextStyle = exports.TextStyle || (exports.TextStyle = {}));
