// @mirohq/miro-api/dist/model/textUpdateRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextUpdateRequest = void 0;
class TextUpdateRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return TextUpdateRequest.attributeTypeMap;
    }
}
exports.TextUpdateRequest = TextUpdateRequest;
/** @ignore */
TextUpdateRequest.discriminator = undefined;
/** @ignore */
TextUpdateRequest.attributeTypeMap = [
    {
        name: 'data',
        baseName: 'data',
        type: 'TextData',
    },
    {
        name: 'style',
        baseName: 'style',
        type: 'TextStyle',
    },
    {
        name: 'position',
        baseName: 'position',
        type: 'PositionChange',
    },
    {
        name: 'geometry',
        baseName: 'geometry',
        type: 'WidthOnlyAdjustableGeometry',
    },
    {
        name: 'parent',
        baseName: 'parent',
        type: 'Parent',
    },
];
