// @mirohq/miro-api/dist/model/tokenInformation.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenInformation = void 0;
class TokenInformation {
    /** @ignore */
    static getAttributeTypeMap() {
        return TokenInformation.attributeTypeMap;
    }
}
exports.TokenInformation = TokenInformation;
/** @ignore */
TokenInformation.discriminator = undefined;
/** @ignore */
TokenInformation.attributeTypeMap = [
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
    {
        name: 'organization',
        baseName: 'organization',
        type: 'OrganizationInformation',
    },
    {
        name: 'team',
        baseName: 'team',
        type: 'TeamInformation',
    },
    {
        name: 'createdBy',
        baseName: 'createdBy',
        type: 'UserInformation',
    },
    {
        name: 'user',
        baseName: 'user',
        type: 'UserInformation',
    },
    {
        name: 'scopes',
        baseName: 'scopes',
        type: 'Array<string>',
    },
];
