// @mirohq/miro-api/dist/model/updateBoardsDataClassificationLabel.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateBoardsDataClassificationLabel = void 0;
class UpdateBoardsDataClassificationLabel {
    constructor() {
        /**
         * Type of the object returned.
         */
        this['type'] = 'boards-data-classification-label-changed';
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return UpdateBoardsDataClassificationLabel.attributeTypeMap;
    }
}
exports.UpdateBoardsDataClassificationLabel = UpdateBoardsDataClassificationLabel;
/** @ignore */
UpdateBoardsDataClassificationLabel.discriminator = undefined;
/** @ignore */
UpdateBoardsDataClassificationLabel.attributeTypeMap = [
    {
        name: 'numberUpdatedBoards',
        baseName: 'numberUpdatedBoards',
        type: 'number',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
