// @mirohq/miro-api/dist/model/updateBoardsDataClassificationLabelRequest.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateBoardsDataClassificationLabelRequest = void 0;
class UpdateBoardsDataClassificationLabelRequest {
    /** @ignore */
    static getAttributeTypeMap() {
        return UpdateBoardsDataClassificationLabelRequest.attributeTypeMap;
    }
}
exports.UpdateBoardsDataClassificationLabelRequest = UpdateBoardsDataClassificationLabelRequest;
/** @ignore */
UpdateBoardsDataClassificationLabelRequest.discriminator = undefined;
/** @ignore */
UpdateBoardsDataClassificationLabelRequest.attributeTypeMap = [
    {
        name: 'labelId',
        baseName: 'labelId',
        type: 'number',
    },
    {
        name: 'notClassifiedOnly',
        baseName: 'notClassifiedOnly',
        type: 'boolean',
    },
];
