// @mirohq/miro-api/dist/model/widgetLinksPlatformExperimentalFeatures.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WidgetLinksPlatformExperimentalFeatures = void 0;
/**
 * @internal
 * Contains applicable links for the item.
 */
class WidgetLinksPlatformExperimentalFeatures {
    /** @ignore */
    static getAttributeTypeMap() {
        return WidgetLinksPlatformExperimentalFeatures.attributeTypeMap;
    }
}
exports.WidgetLinksPlatformExperimentalFeatures = WidgetLinksPlatformExperimentalFeatures;
/** @ignore */
WidgetLinksPlatformExperimentalFeatures.discriminator = undefined;
/** @ignore */
WidgetLinksPlatformExperimentalFeatures.attributeTypeMap = [
    {
        name: 'related',
        baseName: 'related',
        type: 'string',
    },
    {
        name: 'self',
        baseName: 'self',
        type: 'string',
    },
];
