// @mirohq/miro-api/dist/model/widthOnlyAdjustableGeometry.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WidthOnlyAdjustableGeometry = void 0;
/**
 * @internal
 * Contains geometrical information about the item, such as its width or rotation. You can only specify the width of the text item as the height is dynamically updated based on the content.
 */
class WidthOnlyAdjustableGeometry {
    /** @ignore */
    static getAttributeTypeMap() {
        return WidthOnlyAdjustableGeometry.attributeTypeMap;
    }
}
exports.WidthOnlyAdjustableGeometry = WidthOnlyAdjustableGeometry;
/** @ignore */
WidthOnlyAdjustableGeometry.discriminator = undefined;
/** @ignore */
WidthOnlyAdjustableGeometry.attributeTypeMap = [
    {
        name: 'rotation',
        baseName: 'rotation',
        type: 'number',
    },
    {
        name: 'width',
        baseName: 'width',
        type: 'number',
    },
];
